# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: deepvariant/protos/deepvariant.proto

import sys
_b=sys.version_info[0]<3 and (lambda x:x) or (lambda x:x.encode('latin1'))
from google.protobuf.internal import enum_type_wrapper
from google.protobuf import descriptor as _descriptor
from google.protobuf import message as _message
from google.protobuf import reflection as _reflection
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()


from . import realigner_pb2 as deepvariant_dot_protos_dot_realigner__pb2
from . import resources_pb2 as deepvariant_dot_protos_dot_resources__pb2
from .third_party.nucleus.protos import position_pb2 as third__party_dot_nucleus_dot_protos_dot_position__pb2
from .third_party.nucleus.protos import reads_pb2 as third__party_dot_nucleus_dot_protos_dot_reads__pb2
from .third_party.nucleus.protos import variants_pb2 as third__party_dot_nucleus_dot_protos_dot_variants__pb2


DESCRIPTOR = _descriptor.FileDescriptor(
  name='deepvariant/protos/deepvariant.proto',
  package='learning.genomics.deepvariant',
  syntax='proto3',
  serialized_options=None,
  serialized_pb=_b('\n$deepvariant/protos/deepvariant.proto\x12\x1dlearning.genomics.deepvariant\x1a\"deepvariant/protos/realigner.proto\x1a\"deepvariant/protos/resources.proto\x1a)third_party/nucleus/protos/position.proto\x1a&third_party/nucleus/protos/reads.proto\x1a)third_party/nucleus/protos/variants.proto\"w\n\x06\x41llele\x12\r\n\x05\x62\x61ses\x18\x01 \x01(\t\x12\x37\n\x04type\x18\x02 \x01(\x0e\x32).learning.genomics.deepvariant.AlleleType\x12\r\n\x05\x63ount\x18\x03 \x01(\x05\x12\x16\n\x0eis_low_quality\x18\x04 \x01(\x08\"\xe3\x04\n\x0b\x41lleleCount\x12/\n\x08position\x18\x01 \x01(\x0b\x32\x1d.nucleus.genomics.v1.Position\x12\x10\n\x08ref_base\x18\x02 \x01(\t\x12!\n\x19ref_supporting_read_count\x18\x03 \x01(\x05\x12Q\n\x0cread_alleles\x18\x04 \x03(\x0b\x32;.learning.genomics.deepvariant.AlleleCount.ReadAllelesEntry\x12#\n\x1bref_nonconfident_read_count\x18\x05 \x01(\x05\x12U\n\x0esample_alleles\x18\x06 \x03(\x0b\x32=.learning.genomics.deepvariant.AlleleCount.SampleAllelesEntry\x12\x17\n\x0ftrack_ref_reads\x18\x07 \x01(\x08\x1aY\n\x10ReadAllelesEntry\x12\x0b\n\x03key\x18\x01 \x01(\t\x12\x34\n\x05value\x18\x02 \x01(\x0b\x32%.learning.genomics.deepvariant.Allele:\x02\x38\x01\x1a\x41\n\x07\x41lleles\x12\x36\n\x07\x61lleles\x18\x01 \x03(\x0b\x32%.learning.genomics.deepvariant.Allele\x1ah\n\x12SampleAllelesEntry\x12\x0b\n\x03key\x18\x01 \x01(\t\x12\x41\n\x05value\x18\x02 \x01(\x0b\x32\x32.learning.genomics.deepvariant.AlleleCount.Alleles:\x02\x38\x01\"\xb2\x01\n\x12\x41lleleCountSummary\x12\x16\n\x0ereference_name\x18\x01 \x01(\t\x12\x10\n\x08position\x18\x02 \x01(\x03\x12\x10\n\x08ref_base\x18\x03 \x01(\t\x12!\n\x19ref_supporting_read_count\x18\x04 \x01(\x05\x12\x18\n\x10total_read_count\x18\x05 \x01(\x05\x12#\n\x1bref_nonconfident_read_count\x18\x06 \x01(\x05\"\xbe\x07\n\x0f\x44\x65\x65pVariantCall\x12-\n\x07variant\x18\x01 \x01(\x0b\x32\x1c.nucleus.genomics.v1.Variant\x12Y\n\x0e\x61llele_support\x18\x02 \x03(\x0b\x32\x41.learning.genomics.deepvariant.DeepVariantCall.AlleleSupportEntry\x12]\n\x10\x61llele_frequency\x18\x03 \x03(\x0b\x32\x43.learning.genomics.deepvariant.DeepVariantCall.AlleleFrequencyEntry\x12\x13\n\x0bref_support\x18\x04 \x03(\t\x12`\n\x12\x61llele_support_ext\x18\x05 \x03(\x0b\x32\x44.learning.genomics.deepvariant.DeepVariantCall.AlleleSupportExtEntry\x12Z\n\x0fref_support_ext\x18\x06 \x01(\x0b\x32\x41.learning.genomics.deepvariant.DeepVariantCall.SupportingReadsExt\x1a%\n\x0fSupportingReads\x12\x12\n\nread_names\x18\x01 \x03(\t\x1at\n\x12\x41lleleSupportEntry\x12\x0b\n\x03key\x18\x01 \x01(\t\x12M\n\x05value\x18\x02 \x01(\x0b\x32>.learning.genomics.deepvariant.DeepVariantCall.SupportingReads:\x02\x38\x01\x1a\x36\n\x14\x41lleleFrequencyEntry\x12\x0b\n\x03key\x18\x01 \x01(\t\x12\r\n\x05value\x18\x02 \x01(\x02:\x02\x38\x01\x1a\x38\n\x0bReadSupport\x12\x11\n\tread_name\x18\x01 \x01(\t\x12\x16\n\x0eis_low_quality\x18\x02 \x01(\x08\x1a\x64\n\x12SupportingReadsExt\x12N\n\nread_infos\x18\x01 \x03(\x0b\x32:.learning.genomics.deepvariant.DeepVariantCall.ReadSupport\x1az\n\x15\x41lleleSupportExtEntry\x12\x0b\n\x03key\x18\x01 \x01(\t\x12P\n\x05value\x18\x02 \x01(\x0b\x32\x41.learning.genomics.deepvariant.DeepVariantCall.SupportingReadsExt:\x02\x38\x01\"\xc0\x01\n\x14\x41lleleCounterOptions\x12\x16\n\x0epartition_size\x18\x01 \x01(\x05\x12@\n\x11read_requirements\x18\x02 \x01(\x0b\x32%.nucleus.genomics.v1.ReadRequirements\x12\x17\n\x0ftrack_ref_reads\x18\x03 \x01(\x08\x12\x17\n\x0fnormalize_reads\x18\x04 \x01(\x08\x12\x1c\n\x14keep_legacy_behavior\x18\x05 \x01(\x08\"\xd4\x03\n\x12\x43\x61llVariantsOutput\x12-\n\x07variant\x18\x01 \x01(\x0b\x32\x1c.nucleus.genomics.v1.Variant\x12^\n\x12\x61lt_allele_indices\x18\x02 \x01(\x0b\x32\x42.learning.genomics.deepvariant.CallVariantsOutput.AltAlleleIndices\x12\x1e\n\x16genotype_probabilities\x18\x03 \x03(\x01\x12O\n\ndebug_info\x18\x04 \x01(\x0b\x32;.learning.genomics.deepvariant.CallVariantsOutput.DebugInfo\x1a#\n\x10\x41ltAlleleIndices\x12\x0f\n\x07indices\x18\x01 \x03(\x05\x1a\x98\x01\n\tDebugInfo\x12\x17\n\x0fpredicted_label\x18\x01 \x01(\x05\x12\x15\n\rhas_insertion\x18\x02 \x01(\x08\x12\x14\n\x0chas_deletion\x18\x03 \x01(\x08\x12\x0e\n\x06is_snp\x18\x04 \x01(\x08\x12\x12\n\ntrue_label\x18\x05 \x01(\x05\x12\x0e\n\x06logits\x18\x06 \x03(\x01\x12\x11\n\tprelogits\x18\x07 \x03(\x01\"\xf7\x02\n\x14VariantCallerOptions\x12\x16\n\x0emin_count_snps\x18\x01 \x01(\x05\x12\x18\n\x10min_count_indels\x18\x02 \x01(\x05\x12\x19\n\x11min_fraction_snps\x18\x03 \x01(\x02\x12\x1b\n\x13min_fraction_indels\x18\x04 \x01(\x02\x12\x1f\n\x17min_fraction_multiplier\x18\x0c \x01(\x02\x12(\n fraction_reference_sites_to_emit\x18\x05 \x01(\x02\x12\x13\n\x0brandom_seed\x18\x06 \x01(\r\x12\x13\n\x0bsample_name\x18\x07 \x01(\t\x12\x0f\n\x07p_error\x18\x08 \x01(\x02\x12\x0e\n\x06max_gq\x18\t \x01(\x05\x12\x15\n\rgq_resolution\x18\n \x01(\x05\x12\x0e\n\x06ploidy\x18\x0b \x01(\x05\x12\x1f\n\x17skip_uncalled_genotypes\x18\r \x01(\x08\x12\x17\n\x0ftrack_ref_reads\x18\x0e \x01(\x08\"\x17\n\x15VariantLabelerOptions\"\xd7\x0b\n\x12PileupImageOptions\x12\x0e\n\x06height\x18\x01 \x01(\x05\x12\r\n\x05width\x18\x02 \x01(\x05\x12\x1d\n\x15reference_band_height\x18\x03 \x01(\x05\x12!\n\x19\x62\x61se_color_offset_a_and_g\x18\x04 \x01(\x05\x12!\n\x19\x62\x61se_color_offset_t_and_c\x18\x05 \x01(\x05\x12\x19\n\x11\x62\x61se_color_stride\x18\x06 \x01(\x05\x12\x17\n\x0freference_alpha\x18\x07 \x01(\x02\x12\x1e\n\x16reference_base_quality\x18\x08 \x01(\x05\x12$\n\x1c\x61llele_supporting_read_alpha\x18\t \x01(\x02\x12*\n\"other_allele_supporting_read_alpha\x18  \x01(\x02\x12&\n\x1e\x61llele_unsupporting_read_alpha\x18\n \x01(\x02\x12%\n\x1dreference_matching_read_alpha\x18\x0b \x01(\x02\x12(\n reference_mismatching_read_alpha\x18\x0c \x01(\x02\x12!\n\x19indel_anchoring_base_char\x18\r \x01(\t\x12\x1d\n\x15positive_strand_color\x18\x0e \x01(\x05\x12\x1d\n\x15negative_strand_color\x18\x0f \x01(\x05\x12\x18\n\x10\x62\x61se_quality_cap\x18\x10 \x01(\x05\x12\x1e\n\x16read_overlap_buffer_bp\x18\x11 \x01(\x05\x12@\n\x11read_requirements\x18\x12 \x01(\x0b\x32%.nucleus.genomics.v1.ReadRequirements\x12^\n\x12multi_allelic_mode\x18\x13 \x01(\x0e\x32\x42.learning.genomics.deepvariant.PileupImageOptions.MultiAllelicMode\x12\x1b\n\x13mapping_quality_cap\x18\x14 \x01(\x05\x12\x13\n\x0brandom_seed\x18\x15 \x01(\r\x12\x14\n\x0cnum_channels\x18\x16 \x01(\x05\x12\x1f\n\x17unused_insert_base_char\x18\x17 \x01(\t\x12\x1f\n\x17unused_delete_base_char\x18\x18 \x01(\t\x12\"\n\x1aunused_custom_pileup_image\x18\x19 \x01(\x08\x12$\n\x1cunused_sequencing_type_image\x18\x1a \x01(\x08\x12Y\n\x0fsequencing_type\x18\x1b \x01(\x0e\x32@.learning.genomics.deepvariant.PileupImageOptions.SequencingType\x12\x1a\n\x12\x61lt_aligned_pileup\x18\x1e \x01(\t\x12\x1a\n\x12sort_by_haplotypes\x18\x1f \x01(\x08\x12%\n\x1dmin_non_zero_allele_frequency\x18! \x01(\x02\x12\x1c\n\x14use_allele_frequency\x18\" \x01(\x08\x12\x1a\n\x12types_to_alt_align\x18$ \x01(\t\x12\x16\n\x0e\x61\x64\x64_hp_channel\x18% \x01(\x08\x12%\n\x1dhp_tag_for_assembly_polishing\x18& \x01(\x05\x12\x10\n\x08\x63hannels\x18\' \x03(\t\x12,\n sort_by_haplotypes_sample_hp_tag\x18# \x01(\x05\x42\x02\x18\x01\"R\n\x10MultiAllelicMode\x12\x0f\n\x0bUNSPECIFIED\x10\x00\x12\x16\n\x12\x41\x44\x44_HET_ALT_IMAGES\x10\x01\x12\x15\n\x11NO_HET_ALT_IMAGES\x10\x02\"F\n\x0eSequencingType\x12\x18\n\x14UNSPECIFIED_SEQ_TYPE\x10\x00\x12\x07\n\x03WGS\x10\x01\x12\x07\n\x03WES\x10\x02\x12\x08\n\x04TRIO\x10\x03\"\x80\x02\n\rSampleOptions\x12\x0c\n\x04role\x18\x06 \x01(\t\x12\x0c\n\x04name\x18\x07 \x01(\t\x12\x17\n\x0freads_filenames\x18\x01 \x03(\t\x12\x1b\n\x13\x64ownsample_fraction\x18\x02 \x01(\x02\x12S\n\x16variant_caller_options\x18\x03 \x01(\x0b\x32\x33.learning.genomics.deepvariant.VariantCallerOptions\x12\x15\n\rpileup_height\x18\x04 \x01(\x05\x12\r\n\x05order\x18\x05 \x03(\x05\x12\"\n\x1aproposed_variants_filename\x18\x08 \x01(\t\"\xc9\x11\n\x13MakeExamplesOptions\x12\x17\n\x0f\x65xclude_contigs\x18\x01 \x03(\t\x12\x17\n\x0f\x63\x61lling_regions\x18\x02 \x03(\t\x12\x13\n\x0brandom_seed\x18\x03 \x01(\r\x12\x0f\n\x07n_cores\x18\x04 \x01(\x05\x12S\n\x16\x61llele_counter_options\x18\x05 \x01(\x0b\x32\x33.learning.genomics.deepvariant.AlleleCounterOptions\x12^\n!deprecated_variant_caller_options\x18\x06 \x01(\x0b\x32\x33.learning.genomics.deepvariant.VariantCallerOptions\x12\x46\n\x0bpic_options\x18\x07 \x01(\x0b\x32\x31.learning.genomics.deepvariant.PileupImageOptions\x12M\n\x0flabeler_options\x18\x08 \x01(\x0b\x32\x34.learning.genomics.deepvariant.VariantLabelerOptions\x12@\n\x11read_requirements\x18\t \x01(\x0b\x32%.nucleus.genomics.v1.ReadRequirements\x12\x1a\n\x12reference_filename\x18\n \x01(\t\x12\"\n\x1a\x64\x65precated_reads_filenames\x18  \x03(\t\x12!\n\x19\x64\x65precated_reads_filename\x18\x0b \x01(\t\x12\x1b\n\x13\x63\x61ndidates_filename\x18\x0c \x01(\t\x12\x19\n\x11\x65xamples_filename\x18\r \x01(\t\x12\"\n\x1a\x63onfident_regions_filename\x18\x0e \x01(\t\x12\x1f\n\x17truth_variants_filename\x18\x0f \x01(\t\x12\x31\n%deprecated_proposed_variants_filename\x18! \x01(\tB\x02\x18\x01\x12\x15\n\rgvcf_filename\x18\x10 \x01(\t\x12\x16\n\x0einclude_med_dp\x18+ \x01(\x08\x12\x12\n\nmodel_name\x18\x11 \x01(\t\x12\x45\n\x04mode\x18\x12 \x01(\x0e\x32\x37.learning.genomics.deepvariant.MakeExamplesOptions.Mode\x12$\n\x1cmin_shared_contigs_basepairs\x18\x13 \x01(\x02\x12\x0f\n\x07task_id\x18\x14 \x01(\x05\x12\x12\n\nnum_shards\x18\x15 \x01(\x05\x12\x19\n\x11realigner_enabled\x18\x16 \x01(\x08\x12J\n\x11realigner_options\x18\x17 \x01(\x0b\x32/.learning.genomics.deepvariant.RealignerOptions\x12\x1f\n\x17max_reads_per_partition\x18\x18 \x01(\x05\x12&\n\x1e\x64\x65precated_downsample_fraction\x18\x19 \x01(\x02\x12\x1f\n\x17\x65xclude_calling_regions\x18\x1a \x03(\t\x12^\n\x11labeler_algorithm\x18\x1b \x01(\x0e\x32\x43.learning.genomics.deepvariant.MakeExamplesOptions.LabelerAlgorithm\x12\x19\n\x11run_info_filename\x18\x1c \x01(\t\x12#\n\x1buse_original_quality_scores\x18\x1d \x01(\x08\x12\x1c\n\x14select_variant_types\x18\x1e \x03(\t\x12X\n\x0evariant_caller\x18\x1f \x01(\x0e\x32@.learning.genomics.deepvariant.MakeExamplesOptions.VariantCaller\x12\x1c\n\x14use_allele_frequency\x18\" \x01(\x08\x12 \n\x18population_vcf_filenames\x18# \x03(\t\x12\x19\n\x11runtime_by_region\x18$ \x01(\t\x12\x18\n\x10use_ref_for_cram\x18% \x01(\x08\x12\x1c\n\x14parse_sam_aux_fields\x18& \x01(\x08\x12\x1a\n\x12\x61ux_fields_to_keep\x18\x32 \x03(\t\x12\x16\n\x0ehts_block_size\x18\' \x01(\x05\x12\"\n\x1alogging_every_n_candidates\x18( \x01(\x05\x12/\n\'customized_classes_labeler_classes_list\x18) \x01(\t\x12\x32\n*customized_classes_labeler_info_field_name\x18* \x01(\t\x12\x19\n\x11main_sample_index\x18, \x01(\x05\x12\x11\n\tbam_fname\x18- \x01(\t\x12\x44\n\x0esample_options\x18. \x03(\x0b\x32,.learning.genomics.deepvariant.SampleOptions\x12\x1c\n\x14sample_role_to_train\x18/ \x01(\t\"2\n\x04Mode\x12\x0f\n\x0bUNSPECIFIED\x10\x00\x12\x0b\n\x07\x43\x41LLING\x10\x01\x12\x0c\n\x08TRAINING\x10\x02\"\x84\x01\n\x10LabelerAlgorithm\x12!\n\x1dUNSPECIFIED_LABELER_ALGORITHM\x10\x00\x12\x16\n\x12POSITIONAL_LABELER\x10\x01\x12\x15\n\x11HAPLOTYPE_LABELER\x10\x02\x12\x1e\n\x1a\x43USTOMIZED_CLASSES_LABELER\x10\x03\"^\n\rVariantCaller\x12\x16\n\x12UNSPECIFIED_CALLER\x10\x00\x12\x19\n\x15VERY_SENSITIVE_CALLER\x10\x01\x12\x1a\n\x16VCF_CANDIDATE_IMPORTER\x10\x02\"U\n\x18\x44\x65\x65pVariantDatasetConfig\x12\x0c\n\x04name\x18\x01 \x01(\t\x12\x15\n\rtfrecord_path\x18\x02 \x01(\t\x12\x14\n\x0cnum_examples\x18\x03 \x01(\r\"\xba\x04\n\x18\x44\x65\x65pVariantInputDatasets\x12Q\n\x08\x64\x61tasets\x18\x01 \x03(\x0b\x32?.learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset\x12\x12\n\ngroup_name\x18\x02 \x01(\t\x1a\xb6\x03\n\x07\x44\x61taset\x12\x19\n\x11\x63onfident_regions\x18\x01 \x01(\t\x12\r\n\x05reads\x18\x02 \x01(\t\x12\x0b\n\x03ref\x18\x03 \x01(\t\x12\x16\n\x0etruth_variants\x18\x04 \x01(\t\x12\x0f\n\x07regions\x18\x05 \x01(\t\x12&\n\x1etraining_random_emit_ref_sites\x18\x06 \x01(\t\x12\x1c\n\x14\x64ownsample_fractions\x18\x07 \x03(\t\x12\x17\n\x0f\x65xclude_regions\x18\x08 \x01(\t\x12\x13\n\x0bsample_name\x18\t \x01(\t\x12\x17\n\x0fsequencing_type\x18\n \x01(\t\x12\x17\n\x0fpopulation_vcfs\x18\x0b \x01(\t\x12%\n\x1dhp_tag_for_assembly_polishing\x18\r \x01(\x05\x12\x1a\n\x12sort_by_haplotypes\x18\x0e \x01(\x08\x12\x1c\n\x14parse_sam_aux_fields\x18\x0f \x01(\x08\x12\x16\n\x0e\x61\x64\x64_hp_channel\x18\x10 \x01(\x08\x12,\n sort_by_haplotypes_sample_hp_tag\x18\x0c \x01(\x05\x42\x02\x18\x01\"h\n\x1d\x44\x65\x65pVariantInputDatasetsGroup\x12G\n\x06groups\x18\x01 \x03(\x0b\x32\x37.learning.genomics.deepvariant.DeepVariantInputDatasets\"\xf4\x04\n\x0fLabelingMetrics\x12\x1d\n\x15n_truth_variant_sites\x18\x01 \x01(\x05\x12\x1f\n\x17n_truth_variant_alleles\x18\x02 \x01(\x05\x12!\n\x19n_candidate_variant_sites\x18\x03 \x01(\x05\x12#\n\x1bn_candidate_variant_alleles\x18\x04 \x01(\x05\x12/\n\'n_non_confident_candidate_variant_sites\x18\x05 \x01(\x05\x12\x1d\n\x15n_true_positive_sites\x18\x06 \x01(\x05\x12\x1f\n\x17n_true_positive_alleles\x18\x07 \x01(\x05\x12\x1e\n\x16n_false_negative_sites\x18\x08 \x01(\x05\x12 \n\x18n_false_negative_alleles\x18\t \x01(\x05\x12\x1e\n\x16n_false_positive_sites\x18\n \x01(\x05\x12 \n\x18n_false_positive_alleles\x18\x0b \x01(\x05\x12\"\n\x1an_inexact_position_matches\x18\x0c \x01(\x05\x12 \n\x18n_exact_position_matches\x18\r \x01(\x05\x12+\n#n_exact_position_and_allele_matches\x18\x0e \x01(\x05\x12\x38\n0n_exact_position_and_allele_and_genotype_matches\x18\x0f \x01(\x05\x12\x37\n/n_truth_multiallelics_sites_with_missed_alleles\x18\x10 \x01(\x05\"\x8e\x01\n\x11MakeExamplesStats\x12\x14\n\x0cnum_examples\x18\x01 \x01(\x05\x12\x12\n\nnum_indels\x18\x02 \x01(\x05\x12\x10\n\x08num_snps\x18\x03 \x01(\x05\x12\x13\n\x0bnum_class_0\x18\x04 \x01(\x05\x12\x13\n\x0bnum_class_1\x18\x05 \x01(\x05\x12\x13\n\x0bnum_class_2\x18\x06 \x01(\x05\"\xaf\x02\n\x13MakeExamplesRunInfo\x12\x43\n\x07options\x18\x01 \x01(\x0b\x32\x32.learning.genomics.deepvariant.MakeExamplesOptions\x12H\n\x10labeling_metrics\x18\x02 \x01(\x0b\x32..learning.genomics.deepvariant.LabelingMetrics\x12H\n\x10resource_metrics\x18\x03 \x01(\x0b\x32..learning.genomics.deepvariant.ResourceMetrics\x12?\n\x05stats\x18\x04 \x01(\x0b\x32\x30.learning.genomics.deepvariant.MakeExamplesStats*j\n\nAlleleType\x12\x0f\n\x0bUNSPECIFIED\x10\x00\x12\r\n\tREFERENCE\x10\x01\x12\x10\n\x0cSUBSTITUTION\x10\x02\x12\r\n\tINSERTION\x10\x03\x12\x0c\n\x08\x44\x45LETION\x10\x04\x12\r\n\tSOFT_CLIP\x10\x05\x62\x06proto3')
  ,
  dependencies=[deepvariant_dot_protos_dot_realigner__pb2.DESCRIPTOR,deepvariant_dot_protos_dot_resources__pb2.DESCRIPTOR,third__party_dot_nucleus_dot_protos_dot_position__pb2.DESCRIPTOR,third__party_dot_nucleus_dot_protos_dot_reads__pb2.DESCRIPTOR,third__party_dot_nucleus_dot_protos_dot_variants__pb2.DESCRIPTOR,])

_ALLELETYPE = _descriptor.EnumDescriptor(
  name='AlleleType',
  full_name='learning.genomics.deepvariant.AlleleType',
  filename=None,
  file=DESCRIPTOR,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNSPECIFIED', index=0, number=0,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='REFERENCE', index=1, number=1,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='SUBSTITUTION', index=2, number=2,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='INSERTION', index=3, number=3,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='DELETION', index=4, number=4,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='SOFT_CLIP', index=5, number=5,
      serialized_options=None,
      type=None),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=9072,
  serialized_end=9178,
)
_sym_db.RegisterEnumDescriptor(_ALLELETYPE)

AlleleType = enum_type_wrapper.EnumTypeWrapper(_ALLELETYPE)
UNSPECIFIED = 0
REFERENCE = 1
SUBSTITUTION = 2
INSERTION = 3
DELETION = 4
SOFT_CLIP = 5


_PILEUPIMAGEOPTIONS_MULTIALLELICMODE = _descriptor.EnumDescriptor(
  name='MultiAllelicMode',
  full_name='learning.genomics.deepvariant.PileupImageOptions.MultiAllelicMode',
  filename=None,
  file=DESCRIPTOR,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNSPECIFIED', index=0, number=0,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='ADD_HET_ALT_IMAGES', index=1, number=1,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='NO_HET_ALT_IMAGES', index=2, number=2,
      serialized_options=None,
      type=None),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=4557,
  serialized_end=4639,
)
_sym_db.RegisterEnumDescriptor(_PILEUPIMAGEOPTIONS_MULTIALLELICMODE)

_PILEUPIMAGEOPTIONS_SEQUENCINGTYPE = _descriptor.EnumDescriptor(
  name='SequencingType',
  full_name='learning.genomics.deepvariant.PileupImageOptions.SequencingType',
  filename=None,
  file=DESCRIPTOR,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNSPECIFIED_SEQ_TYPE', index=0, number=0,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='WGS', index=1, number=1,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='WES', index=2, number=2,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='TRIO', index=3, number=3,
      serialized_options=None,
      type=None),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=4641,
  serialized_end=4711,
)
_sym_db.RegisterEnumDescriptor(_PILEUPIMAGEOPTIONS_SEQUENCINGTYPE)

_MAKEEXAMPLESOPTIONS_MODE = _descriptor.EnumDescriptor(
  name='Mode',
  full_name='learning.genomics.deepvariant.MakeExamplesOptions.Mode',
  filename=None,
  file=DESCRIPTOR,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNSPECIFIED', index=0, number=0,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='CALLING', index=1, number=1,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='TRAINING', index=2, number=2,
      serialized_options=None,
      type=None),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=6941,
  serialized_end=6991,
)
_sym_db.RegisterEnumDescriptor(_MAKEEXAMPLESOPTIONS_MODE)

_MAKEEXAMPLESOPTIONS_LABELERALGORITHM = _descriptor.EnumDescriptor(
  name='LabelerAlgorithm',
  full_name='learning.genomics.deepvariant.MakeExamplesOptions.LabelerAlgorithm',
  filename=None,
  file=DESCRIPTOR,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNSPECIFIED_LABELER_ALGORITHM', index=0, number=0,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='POSITIONAL_LABELER', index=1, number=1,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='HAPLOTYPE_LABELER', index=2, number=2,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='CUSTOMIZED_CLASSES_LABELER', index=3, number=3,
      serialized_options=None,
      type=None),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=6994,
  serialized_end=7126,
)
_sym_db.RegisterEnumDescriptor(_MAKEEXAMPLESOPTIONS_LABELERALGORITHM)

_MAKEEXAMPLESOPTIONS_VARIANTCALLER = _descriptor.EnumDescriptor(
  name='VariantCaller',
  full_name='learning.genomics.deepvariant.MakeExamplesOptions.VariantCaller',
  filename=None,
  file=DESCRIPTOR,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNSPECIFIED_CALLER', index=0, number=0,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='VERY_SENSITIVE_CALLER', index=1, number=1,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='VCF_CANDIDATE_IMPORTER', index=2, number=2,
      serialized_options=None,
      type=None),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=7128,
  serialized_end=7222,
)
_sym_db.RegisterEnumDescriptor(_MAKEEXAMPLESOPTIONS_VARIANTCALLER)


_ALLELE = _descriptor.Descriptor(
  name='Allele',
  full_name='learning.genomics.deepvariant.Allele',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='bases', full_name='learning.genomics.deepvariant.Allele.bases', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='type', full_name='learning.genomics.deepvariant.Allele.type', index=1,
      number=2, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='count', full_name='learning.genomics.deepvariant.Allele.count', index=2,
      number=3, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='is_low_quality', full_name='learning.genomics.deepvariant.Allele.is_low_quality', index=3,
      number=4, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=269,
  serialized_end=388,
)


_ALLELECOUNT_READALLELESENTRY = _descriptor.Descriptor(
  name='ReadAllelesEntry',
  full_name='learning.genomics.deepvariant.AlleleCount.ReadAllelesEntry',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='key', full_name='learning.genomics.deepvariant.AlleleCount.ReadAllelesEntry.key', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='value', full_name='learning.genomics.deepvariant.AlleleCount.ReadAllelesEntry.value', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=_b('8\001'),
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=740,
  serialized_end=829,
)

_ALLELECOUNT_ALLELES = _descriptor.Descriptor(
  name='Alleles',
  full_name='learning.genomics.deepvariant.AlleleCount.Alleles',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='alleles', full_name='learning.genomics.deepvariant.AlleleCount.Alleles.alleles', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=831,
  serialized_end=896,
)

_ALLELECOUNT_SAMPLEALLELESENTRY = _descriptor.Descriptor(
  name='SampleAllelesEntry',
  full_name='learning.genomics.deepvariant.AlleleCount.SampleAllelesEntry',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='key', full_name='learning.genomics.deepvariant.AlleleCount.SampleAllelesEntry.key', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='value', full_name='learning.genomics.deepvariant.AlleleCount.SampleAllelesEntry.value', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=_b('8\001'),
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=898,
  serialized_end=1002,
)

_ALLELECOUNT = _descriptor.Descriptor(
  name='AlleleCount',
  full_name='learning.genomics.deepvariant.AlleleCount',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='position', full_name='learning.genomics.deepvariant.AlleleCount.position', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='ref_base', full_name='learning.genomics.deepvariant.AlleleCount.ref_base', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='ref_supporting_read_count', full_name='learning.genomics.deepvariant.AlleleCount.ref_supporting_read_count', index=2,
      number=3, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='read_alleles', full_name='learning.genomics.deepvariant.AlleleCount.read_alleles', index=3,
      number=4, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='ref_nonconfident_read_count', full_name='learning.genomics.deepvariant.AlleleCount.ref_nonconfident_read_count', index=4,
      number=5, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='sample_alleles', full_name='learning.genomics.deepvariant.AlleleCount.sample_alleles', index=5,
      number=6, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='track_ref_reads', full_name='learning.genomics.deepvariant.AlleleCount.track_ref_reads', index=6,
      number=7, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[_ALLELECOUNT_READALLELESENTRY, _ALLELECOUNT_ALLELES, _ALLELECOUNT_SAMPLEALLELESENTRY, ],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=391,
  serialized_end=1002,
)


_ALLELECOUNTSUMMARY = _descriptor.Descriptor(
  name='AlleleCountSummary',
  full_name='learning.genomics.deepvariant.AlleleCountSummary',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='reference_name', full_name='learning.genomics.deepvariant.AlleleCountSummary.reference_name', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='position', full_name='learning.genomics.deepvariant.AlleleCountSummary.position', index=1,
      number=2, type=3, cpp_type=2, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='ref_base', full_name='learning.genomics.deepvariant.AlleleCountSummary.ref_base', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='ref_supporting_read_count', full_name='learning.genomics.deepvariant.AlleleCountSummary.ref_supporting_read_count', index=3,
      number=4, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='total_read_count', full_name='learning.genomics.deepvariant.AlleleCountSummary.total_read_count', index=4,
      number=5, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='ref_nonconfident_read_count', full_name='learning.genomics.deepvariant.AlleleCountSummary.ref_nonconfident_read_count', index=5,
      number=6, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=1005,
  serialized_end=1183,
)


_DEEPVARIANTCALL_SUPPORTINGREADS = _descriptor.Descriptor(
  name='SupportingReads',
  full_name='learning.genomics.deepvariant.DeepVariantCall.SupportingReads',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='read_names', full_name='learning.genomics.deepvariant.DeepVariantCall.SupportingReads.read_names', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=1649,
  serialized_end=1686,
)

_DEEPVARIANTCALL_ALLELESUPPORTENTRY = _descriptor.Descriptor(
  name='AlleleSupportEntry',
  full_name='learning.genomics.deepvariant.DeepVariantCall.AlleleSupportEntry',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='key', full_name='learning.genomics.deepvariant.DeepVariantCall.AlleleSupportEntry.key', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='value', full_name='learning.genomics.deepvariant.DeepVariantCall.AlleleSupportEntry.value', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=_b('8\001'),
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=1688,
  serialized_end=1804,
)

_DEEPVARIANTCALL_ALLELEFREQUENCYENTRY = _descriptor.Descriptor(
  name='AlleleFrequencyEntry',
  full_name='learning.genomics.deepvariant.DeepVariantCall.AlleleFrequencyEntry',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='key', full_name='learning.genomics.deepvariant.DeepVariantCall.AlleleFrequencyEntry.key', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='value', full_name='learning.genomics.deepvariant.DeepVariantCall.AlleleFrequencyEntry.value', index=1,
      number=2, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=_b('8\001'),
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=1806,
  serialized_end=1860,
)

_DEEPVARIANTCALL_READSUPPORT = _descriptor.Descriptor(
  name='ReadSupport',
  full_name='learning.genomics.deepvariant.DeepVariantCall.ReadSupport',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='read_name', full_name='learning.genomics.deepvariant.DeepVariantCall.ReadSupport.read_name', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='is_low_quality', full_name='learning.genomics.deepvariant.DeepVariantCall.ReadSupport.is_low_quality', index=1,
      number=2, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=1862,
  serialized_end=1918,
)

_DEEPVARIANTCALL_SUPPORTINGREADSEXT = _descriptor.Descriptor(
  name='SupportingReadsExt',
  full_name='learning.genomics.deepvariant.DeepVariantCall.SupportingReadsExt',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='read_infos', full_name='learning.genomics.deepvariant.DeepVariantCall.SupportingReadsExt.read_infos', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=1920,
  serialized_end=2020,
)

_DEEPVARIANTCALL_ALLELESUPPORTEXTENTRY = _descriptor.Descriptor(
  name='AlleleSupportExtEntry',
  full_name='learning.genomics.deepvariant.DeepVariantCall.AlleleSupportExtEntry',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='key', full_name='learning.genomics.deepvariant.DeepVariantCall.AlleleSupportExtEntry.key', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='value', full_name='learning.genomics.deepvariant.DeepVariantCall.AlleleSupportExtEntry.value', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=_b('8\001'),
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=2022,
  serialized_end=2144,
)

_DEEPVARIANTCALL = _descriptor.Descriptor(
  name='DeepVariantCall',
  full_name='learning.genomics.deepvariant.DeepVariantCall',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='variant', full_name='learning.genomics.deepvariant.DeepVariantCall.variant', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='allele_support', full_name='learning.genomics.deepvariant.DeepVariantCall.allele_support', index=1,
      number=2, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='allele_frequency', full_name='learning.genomics.deepvariant.DeepVariantCall.allele_frequency', index=2,
      number=3, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='ref_support', full_name='learning.genomics.deepvariant.DeepVariantCall.ref_support', index=3,
      number=4, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='allele_support_ext', full_name='learning.genomics.deepvariant.DeepVariantCall.allele_support_ext', index=4,
      number=5, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='ref_support_ext', full_name='learning.genomics.deepvariant.DeepVariantCall.ref_support_ext', index=5,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[_DEEPVARIANTCALL_SUPPORTINGREADS, _DEEPVARIANTCALL_ALLELESUPPORTENTRY, _DEEPVARIANTCALL_ALLELEFREQUENCYENTRY, _DEEPVARIANTCALL_READSUPPORT, _DEEPVARIANTCALL_SUPPORTINGREADSEXT, _DEEPVARIANTCALL_ALLELESUPPORTEXTENTRY, ],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=1186,
  serialized_end=2144,
)


_ALLELECOUNTEROPTIONS = _descriptor.Descriptor(
  name='AlleleCounterOptions',
  full_name='learning.genomics.deepvariant.AlleleCounterOptions',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='partition_size', full_name='learning.genomics.deepvariant.AlleleCounterOptions.partition_size', index=0,
      number=1, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='read_requirements', full_name='learning.genomics.deepvariant.AlleleCounterOptions.read_requirements', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='track_ref_reads', full_name='learning.genomics.deepvariant.AlleleCounterOptions.track_ref_reads', index=2,
      number=3, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='normalize_reads', full_name='learning.genomics.deepvariant.AlleleCounterOptions.normalize_reads', index=3,
      number=4, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='keep_legacy_behavior', full_name='learning.genomics.deepvariant.AlleleCounterOptions.keep_legacy_behavior', index=4,
      number=5, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=2147,
  serialized_end=2339,
)


_CALLVARIANTSOUTPUT_ALTALLELEINDICES = _descriptor.Descriptor(
  name='AltAlleleIndices',
  full_name='learning.genomics.deepvariant.CallVariantsOutput.AltAlleleIndices',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='indices', full_name='learning.genomics.deepvariant.CallVariantsOutput.AltAlleleIndices.indices', index=0,
      number=1, type=5, cpp_type=1, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=2620,
  serialized_end=2655,
)

_CALLVARIANTSOUTPUT_DEBUGINFO = _descriptor.Descriptor(
  name='DebugInfo',
  full_name='learning.genomics.deepvariant.CallVariantsOutput.DebugInfo',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='predicted_label', full_name='learning.genomics.deepvariant.CallVariantsOutput.DebugInfo.predicted_label', index=0,
      number=1, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='has_insertion', full_name='learning.genomics.deepvariant.CallVariantsOutput.DebugInfo.has_insertion', index=1,
      number=2, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='has_deletion', full_name='learning.genomics.deepvariant.CallVariantsOutput.DebugInfo.has_deletion', index=2,
      number=3, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='is_snp', full_name='learning.genomics.deepvariant.CallVariantsOutput.DebugInfo.is_snp', index=3,
      number=4, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='true_label', full_name='learning.genomics.deepvariant.CallVariantsOutput.DebugInfo.true_label', index=4,
      number=5, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='logits', full_name='learning.genomics.deepvariant.CallVariantsOutput.DebugInfo.logits', index=5,
      number=6, type=1, cpp_type=5, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='prelogits', full_name='learning.genomics.deepvariant.CallVariantsOutput.DebugInfo.prelogits', index=6,
      number=7, type=1, cpp_type=5, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=2658,
  serialized_end=2810,
)

_CALLVARIANTSOUTPUT = _descriptor.Descriptor(
  name='CallVariantsOutput',
  full_name='learning.genomics.deepvariant.CallVariantsOutput',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='variant', full_name='learning.genomics.deepvariant.CallVariantsOutput.variant', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='alt_allele_indices', full_name='learning.genomics.deepvariant.CallVariantsOutput.alt_allele_indices', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='genotype_probabilities', full_name='learning.genomics.deepvariant.CallVariantsOutput.genotype_probabilities', index=2,
      number=3, type=1, cpp_type=5, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='debug_info', full_name='learning.genomics.deepvariant.CallVariantsOutput.debug_info', index=3,
      number=4, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[_CALLVARIANTSOUTPUT_ALTALLELEINDICES, _CALLVARIANTSOUTPUT_DEBUGINFO, ],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=2342,
  serialized_end=2810,
)


_VARIANTCALLEROPTIONS = _descriptor.Descriptor(
  name='VariantCallerOptions',
  full_name='learning.genomics.deepvariant.VariantCallerOptions',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='min_count_snps', full_name='learning.genomics.deepvariant.VariantCallerOptions.min_count_snps', index=0,
      number=1, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='min_count_indels', full_name='learning.genomics.deepvariant.VariantCallerOptions.min_count_indels', index=1,
      number=2, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='min_fraction_snps', full_name='learning.genomics.deepvariant.VariantCallerOptions.min_fraction_snps', index=2,
      number=3, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='min_fraction_indels', full_name='learning.genomics.deepvariant.VariantCallerOptions.min_fraction_indels', index=3,
      number=4, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='min_fraction_multiplier', full_name='learning.genomics.deepvariant.VariantCallerOptions.min_fraction_multiplier', index=4,
      number=12, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='fraction_reference_sites_to_emit', full_name='learning.genomics.deepvariant.VariantCallerOptions.fraction_reference_sites_to_emit', index=5,
      number=5, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='random_seed', full_name='learning.genomics.deepvariant.VariantCallerOptions.random_seed', index=6,
      number=6, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='sample_name', full_name='learning.genomics.deepvariant.VariantCallerOptions.sample_name', index=7,
      number=7, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='p_error', full_name='learning.genomics.deepvariant.VariantCallerOptions.p_error', index=8,
      number=8, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='max_gq', full_name='learning.genomics.deepvariant.VariantCallerOptions.max_gq', index=9,
      number=9, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='gq_resolution', full_name='learning.genomics.deepvariant.VariantCallerOptions.gq_resolution', index=10,
      number=10, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='ploidy', full_name='learning.genomics.deepvariant.VariantCallerOptions.ploidy', index=11,
      number=11, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='skip_uncalled_genotypes', full_name='learning.genomics.deepvariant.VariantCallerOptions.skip_uncalled_genotypes', index=12,
      number=13, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='track_ref_reads', full_name='learning.genomics.deepvariant.VariantCallerOptions.track_ref_reads', index=13,
      number=14, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=2813,
  serialized_end=3188,
)


_VARIANTLABELEROPTIONS = _descriptor.Descriptor(
  name='VariantLabelerOptions',
  full_name='learning.genomics.deepvariant.VariantLabelerOptions',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=3190,
  serialized_end=3213,
)


_PILEUPIMAGEOPTIONS = _descriptor.Descriptor(
  name='PileupImageOptions',
  full_name='learning.genomics.deepvariant.PileupImageOptions',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='height', full_name='learning.genomics.deepvariant.PileupImageOptions.height', index=0,
      number=1, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='width', full_name='learning.genomics.deepvariant.PileupImageOptions.width', index=1,
      number=2, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='reference_band_height', full_name='learning.genomics.deepvariant.PileupImageOptions.reference_band_height', index=2,
      number=3, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='base_color_offset_a_and_g', full_name='learning.genomics.deepvariant.PileupImageOptions.base_color_offset_a_and_g', index=3,
      number=4, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='base_color_offset_t_and_c', full_name='learning.genomics.deepvariant.PileupImageOptions.base_color_offset_t_and_c', index=4,
      number=5, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='base_color_stride', full_name='learning.genomics.deepvariant.PileupImageOptions.base_color_stride', index=5,
      number=6, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='reference_alpha', full_name='learning.genomics.deepvariant.PileupImageOptions.reference_alpha', index=6,
      number=7, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='reference_base_quality', full_name='learning.genomics.deepvariant.PileupImageOptions.reference_base_quality', index=7,
      number=8, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='allele_supporting_read_alpha', full_name='learning.genomics.deepvariant.PileupImageOptions.allele_supporting_read_alpha', index=8,
      number=9, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='other_allele_supporting_read_alpha', full_name='learning.genomics.deepvariant.PileupImageOptions.other_allele_supporting_read_alpha', index=9,
      number=32, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='allele_unsupporting_read_alpha', full_name='learning.genomics.deepvariant.PileupImageOptions.allele_unsupporting_read_alpha', index=10,
      number=10, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='reference_matching_read_alpha', full_name='learning.genomics.deepvariant.PileupImageOptions.reference_matching_read_alpha', index=11,
      number=11, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='reference_mismatching_read_alpha', full_name='learning.genomics.deepvariant.PileupImageOptions.reference_mismatching_read_alpha', index=12,
      number=12, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='indel_anchoring_base_char', full_name='learning.genomics.deepvariant.PileupImageOptions.indel_anchoring_base_char', index=13,
      number=13, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='positive_strand_color', full_name='learning.genomics.deepvariant.PileupImageOptions.positive_strand_color', index=14,
      number=14, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='negative_strand_color', full_name='learning.genomics.deepvariant.PileupImageOptions.negative_strand_color', index=15,
      number=15, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='base_quality_cap', full_name='learning.genomics.deepvariant.PileupImageOptions.base_quality_cap', index=16,
      number=16, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='read_overlap_buffer_bp', full_name='learning.genomics.deepvariant.PileupImageOptions.read_overlap_buffer_bp', index=17,
      number=17, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='read_requirements', full_name='learning.genomics.deepvariant.PileupImageOptions.read_requirements', index=18,
      number=18, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='multi_allelic_mode', full_name='learning.genomics.deepvariant.PileupImageOptions.multi_allelic_mode', index=19,
      number=19, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='mapping_quality_cap', full_name='learning.genomics.deepvariant.PileupImageOptions.mapping_quality_cap', index=20,
      number=20, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='random_seed', full_name='learning.genomics.deepvariant.PileupImageOptions.random_seed', index=21,
      number=21, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='num_channels', full_name='learning.genomics.deepvariant.PileupImageOptions.num_channels', index=22,
      number=22, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='unused_insert_base_char', full_name='learning.genomics.deepvariant.PileupImageOptions.unused_insert_base_char', index=23,
      number=23, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='unused_delete_base_char', full_name='learning.genomics.deepvariant.PileupImageOptions.unused_delete_base_char', index=24,
      number=24, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='unused_custom_pileup_image', full_name='learning.genomics.deepvariant.PileupImageOptions.unused_custom_pileup_image', index=25,
      number=25, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='unused_sequencing_type_image', full_name='learning.genomics.deepvariant.PileupImageOptions.unused_sequencing_type_image', index=26,
      number=26, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='sequencing_type', full_name='learning.genomics.deepvariant.PileupImageOptions.sequencing_type', index=27,
      number=27, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='alt_aligned_pileup', full_name='learning.genomics.deepvariant.PileupImageOptions.alt_aligned_pileup', index=28,
      number=30, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='sort_by_haplotypes', full_name='learning.genomics.deepvariant.PileupImageOptions.sort_by_haplotypes', index=29,
      number=31, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='min_non_zero_allele_frequency', full_name='learning.genomics.deepvariant.PileupImageOptions.min_non_zero_allele_frequency', index=30,
      number=33, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='use_allele_frequency', full_name='learning.genomics.deepvariant.PileupImageOptions.use_allele_frequency', index=31,
      number=34, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='types_to_alt_align', full_name='learning.genomics.deepvariant.PileupImageOptions.types_to_alt_align', index=32,
      number=36, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='add_hp_channel', full_name='learning.genomics.deepvariant.PileupImageOptions.add_hp_channel', index=33,
      number=37, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='hp_tag_for_assembly_polishing', full_name='learning.genomics.deepvariant.PileupImageOptions.hp_tag_for_assembly_polishing', index=34,
      number=38, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='channels', full_name='learning.genomics.deepvariant.PileupImageOptions.channels', index=35,
      number=39, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='sort_by_haplotypes_sample_hp_tag', full_name='learning.genomics.deepvariant.PileupImageOptions.sort_by_haplotypes_sample_hp_tag', index=36,
      number=35, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=_b('\030\001'), file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _PILEUPIMAGEOPTIONS_MULTIALLELICMODE,
    _PILEUPIMAGEOPTIONS_SEQUENCINGTYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=3216,
  serialized_end=4711,
)


_SAMPLEOPTIONS = _descriptor.Descriptor(
  name='SampleOptions',
  full_name='learning.genomics.deepvariant.SampleOptions',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='role', full_name='learning.genomics.deepvariant.SampleOptions.role', index=0,
      number=6, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='name', full_name='learning.genomics.deepvariant.SampleOptions.name', index=1,
      number=7, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='reads_filenames', full_name='learning.genomics.deepvariant.SampleOptions.reads_filenames', index=2,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='downsample_fraction', full_name='learning.genomics.deepvariant.SampleOptions.downsample_fraction', index=3,
      number=2, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='variant_caller_options', full_name='learning.genomics.deepvariant.SampleOptions.variant_caller_options', index=4,
      number=3, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='pileup_height', full_name='learning.genomics.deepvariant.SampleOptions.pileup_height', index=5,
      number=4, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='order', full_name='learning.genomics.deepvariant.SampleOptions.order', index=6,
      number=5, type=5, cpp_type=1, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='proposed_variants_filename', full_name='learning.genomics.deepvariant.SampleOptions.proposed_variants_filename', index=7,
      number=8, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=4714,
  serialized_end=4970,
)


_MAKEEXAMPLESOPTIONS = _descriptor.Descriptor(
  name='MakeExamplesOptions',
  full_name='learning.genomics.deepvariant.MakeExamplesOptions',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='exclude_contigs', full_name='learning.genomics.deepvariant.MakeExamplesOptions.exclude_contigs', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='calling_regions', full_name='learning.genomics.deepvariant.MakeExamplesOptions.calling_regions', index=1,
      number=2, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='random_seed', full_name='learning.genomics.deepvariant.MakeExamplesOptions.random_seed', index=2,
      number=3, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='n_cores', full_name='learning.genomics.deepvariant.MakeExamplesOptions.n_cores', index=3,
      number=4, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='allele_counter_options', full_name='learning.genomics.deepvariant.MakeExamplesOptions.allele_counter_options', index=4,
      number=5, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='deprecated_variant_caller_options', full_name='learning.genomics.deepvariant.MakeExamplesOptions.deprecated_variant_caller_options', index=5,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='pic_options', full_name='learning.genomics.deepvariant.MakeExamplesOptions.pic_options', index=6,
      number=7, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='labeler_options', full_name='learning.genomics.deepvariant.MakeExamplesOptions.labeler_options', index=7,
      number=8, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='read_requirements', full_name='learning.genomics.deepvariant.MakeExamplesOptions.read_requirements', index=8,
      number=9, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='reference_filename', full_name='learning.genomics.deepvariant.MakeExamplesOptions.reference_filename', index=9,
      number=10, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='deprecated_reads_filenames', full_name='learning.genomics.deepvariant.MakeExamplesOptions.deprecated_reads_filenames', index=10,
      number=32, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='deprecated_reads_filename', full_name='learning.genomics.deepvariant.MakeExamplesOptions.deprecated_reads_filename', index=11,
      number=11, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='candidates_filename', full_name='learning.genomics.deepvariant.MakeExamplesOptions.candidates_filename', index=12,
      number=12, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='examples_filename', full_name='learning.genomics.deepvariant.MakeExamplesOptions.examples_filename', index=13,
      number=13, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='confident_regions_filename', full_name='learning.genomics.deepvariant.MakeExamplesOptions.confident_regions_filename', index=14,
      number=14, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='truth_variants_filename', full_name='learning.genomics.deepvariant.MakeExamplesOptions.truth_variants_filename', index=15,
      number=15, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='deprecated_proposed_variants_filename', full_name='learning.genomics.deepvariant.MakeExamplesOptions.deprecated_proposed_variants_filename', index=16,
      number=33, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=_b('\030\001'), file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='gvcf_filename', full_name='learning.genomics.deepvariant.MakeExamplesOptions.gvcf_filename', index=17,
      number=16, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='include_med_dp', full_name='learning.genomics.deepvariant.MakeExamplesOptions.include_med_dp', index=18,
      number=43, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='model_name', full_name='learning.genomics.deepvariant.MakeExamplesOptions.model_name', index=19,
      number=17, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='mode', full_name='learning.genomics.deepvariant.MakeExamplesOptions.mode', index=20,
      number=18, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='min_shared_contigs_basepairs', full_name='learning.genomics.deepvariant.MakeExamplesOptions.min_shared_contigs_basepairs', index=21,
      number=19, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='task_id', full_name='learning.genomics.deepvariant.MakeExamplesOptions.task_id', index=22,
      number=20, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='num_shards', full_name='learning.genomics.deepvariant.MakeExamplesOptions.num_shards', index=23,
      number=21, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='realigner_enabled', full_name='learning.genomics.deepvariant.MakeExamplesOptions.realigner_enabled', index=24,
      number=22, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='realigner_options', full_name='learning.genomics.deepvariant.MakeExamplesOptions.realigner_options', index=25,
      number=23, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='max_reads_per_partition', full_name='learning.genomics.deepvariant.MakeExamplesOptions.max_reads_per_partition', index=26,
      number=24, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='deprecated_downsample_fraction', full_name='learning.genomics.deepvariant.MakeExamplesOptions.deprecated_downsample_fraction', index=27,
      number=25, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='exclude_calling_regions', full_name='learning.genomics.deepvariant.MakeExamplesOptions.exclude_calling_regions', index=28,
      number=26, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='labeler_algorithm', full_name='learning.genomics.deepvariant.MakeExamplesOptions.labeler_algorithm', index=29,
      number=27, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='run_info_filename', full_name='learning.genomics.deepvariant.MakeExamplesOptions.run_info_filename', index=30,
      number=28, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='use_original_quality_scores', full_name='learning.genomics.deepvariant.MakeExamplesOptions.use_original_quality_scores', index=31,
      number=29, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='select_variant_types', full_name='learning.genomics.deepvariant.MakeExamplesOptions.select_variant_types', index=32,
      number=30, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='variant_caller', full_name='learning.genomics.deepvariant.MakeExamplesOptions.variant_caller', index=33,
      number=31, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='use_allele_frequency', full_name='learning.genomics.deepvariant.MakeExamplesOptions.use_allele_frequency', index=34,
      number=34, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='population_vcf_filenames', full_name='learning.genomics.deepvariant.MakeExamplesOptions.population_vcf_filenames', index=35,
      number=35, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='runtime_by_region', full_name='learning.genomics.deepvariant.MakeExamplesOptions.runtime_by_region', index=36,
      number=36, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='use_ref_for_cram', full_name='learning.genomics.deepvariant.MakeExamplesOptions.use_ref_for_cram', index=37,
      number=37, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='parse_sam_aux_fields', full_name='learning.genomics.deepvariant.MakeExamplesOptions.parse_sam_aux_fields', index=38,
      number=38, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='aux_fields_to_keep', full_name='learning.genomics.deepvariant.MakeExamplesOptions.aux_fields_to_keep', index=39,
      number=50, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='hts_block_size', full_name='learning.genomics.deepvariant.MakeExamplesOptions.hts_block_size', index=40,
      number=39, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='logging_every_n_candidates', full_name='learning.genomics.deepvariant.MakeExamplesOptions.logging_every_n_candidates', index=41,
      number=40, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='customized_classes_labeler_classes_list', full_name='learning.genomics.deepvariant.MakeExamplesOptions.customized_classes_labeler_classes_list', index=42,
      number=41, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='customized_classes_labeler_info_field_name', full_name='learning.genomics.deepvariant.MakeExamplesOptions.customized_classes_labeler_info_field_name', index=43,
      number=42, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='main_sample_index', full_name='learning.genomics.deepvariant.MakeExamplesOptions.main_sample_index', index=44,
      number=44, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='bam_fname', full_name='learning.genomics.deepvariant.MakeExamplesOptions.bam_fname', index=45,
      number=45, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='sample_options', full_name='learning.genomics.deepvariant.MakeExamplesOptions.sample_options', index=46,
      number=46, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='sample_role_to_train', full_name='learning.genomics.deepvariant.MakeExamplesOptions.sample_role_to_train', index=47,
      number=47, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _MAKEEXAMPLESOPTIONS_MODE,
    _MAKEEXAMPLESOPTIONS_LABELERALGORITHM,
    _MAKEEXAMPLESOPTIONS_VARIANTCALLER,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=4973,
  serialized_end=7222,
)


_DEEPVARIANTDATASETCONFIG = _descriptor.Descriptor(
  name='DeepVariantDatasetConfig',
  full_name='learning.genomics.deepvariant.DeepVariantDatasetConfig',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='name', full_name='learning.genomics.deepvariant.DeepVariantDatasetConfig.name', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='tfrecord_path', full_name='learning.genomics.deepvariant.DeepVariantDatasetConfig.tfrecord_path', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='num_examples', full_name='learning.genomics.deepvariant.DeepVariantDatasetConfig.num_examples', index=2,
      number=3, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=7224,
  serialized_end=7309,
)


_DEEPVARIANTINPUTDATASETS_DATASET = _descriptor.Descriptor(
  name='Dataset',
  full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='confident_regions', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset.confident_regions', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='reads', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset.reads', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='ref', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset.ref', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='truth_variants', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset.truth_variants', index=3,
      number=4, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='regions', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset.regions', index=4,
      number=5, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='training_random_emit_ref_sites', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset.training_random_emit_ref_sites', index=5,
      number=6, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='downsample_fractions', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset.downsample_fractions', index=6,
      number=7, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='exclude_regions', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset.exclude_regions', index=7,
      number=8, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='sample_name', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset.sample_name', index=8,
      number=9, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='sequencing_type', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset.sequencing_type', index=9,
      number=10, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='population_vcfs', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset.population_vcfs', index=10,
      number=11, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='hp_tag_for_assembly_polishing', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset.hp_tag_for_assembly_polishing', index=11,
      number=13, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='sort_by_haplotypes', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset.sort_by_haplotypes', index=12,
      number=14, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='parse_sam_aux_fields', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset.parse_sam_aux_fields', index=13,
      number=15, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='add_hp_channel', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset.add_hp_channel', index=14,
      number=16, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='sort_by_haplotypes_sample_hp_tag', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset.sort_by_haplotypes_sample_hp_tag', index=15,
      number=12, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=_b('\030\001'), file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=7444,
  serialized_end=7882,
)

_DEEPVARIANTINPUTDATASETS = _descriptor.Descriptor(
  name='DeepVariantInputDatasets',
  full_name='learning.genomics.deepvariant.DeepVariantInputDatasets',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='datasets', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.datasets', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='group_name', full_name='learning.genomics.deepvariant.DeepVariantInputDatasets.group_name', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[_DEEPVARIANTINPUTDATASETS_DATASET, ],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=7312,
  serialized_end=7882,
)


_DEEPVARIANTINPUTDATASETSGROUP = _descriptor.Descriptor(
  name='DeepVariantInputDatasetsGroup',
  full_name='learning.genomics.deepvariant.DeepVariantInputDatasetsGroup',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='groups', full_name='learning.genomics.deepvariant.DeepVariantInputDatasetsGroup.groups', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=7884,
  serialized_end=7988,
)


_LABELINGMETRICS = _descriptor.Descriptor(
  name='LabelingMetrics',
  full_name='learning.genomics.deepvariant.LabelingMetrics',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='n_truth_variant_sites', full_name='learning.genomics.deepvariant.LabelingMetrics.n_truth_variant_sites', index=0,
      number=1, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='n_truth_variant_alleles', full_name='learning.genomics.deepvariant.LabelingMetrics.n_truth_variant_alleles', index=1,
      number=2, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='n_candidate_variant_sites', full_name='learning.genomics.deepvariant.LabelingMetrics.n_candidate_variant_sites', index=2,
      number=3, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='n_candidate_variant_alleles', full_name='learning.genomics.deepvariant.LabelingMetrics.n_candidate_variant_alleles', index=3,
      number=4, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='n_non_confident_candidate_variant_sites', full_name='learning.genomics.deepvariant.LabelingMetrics.n_non_confident_candidate_variant_sites', index=4,
      number=5, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='n_true_positive_sites', full_name='learning.genomics.deepvariant.LabelingMetrics.n_true_positive_sites', index=5,
      number=6, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='n_true_positive_alleles', full_name='learning.genomics.deepvariant.LabelingMetrics.n_true_positive_alleles', index=6,
      number=7, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='n_false_negative_sites', full_name='learning.genomics.deepvariant.LabelingMetrics.n_false_negative_sites', index=7,
      number=8, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='n_false_negative_alleles', full_name='learning.genomics.deepvariant.LabelingMetrics.n_false_negative_alleles', index=8,
      number=9, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='n_false_positive_sites', full_name='learning.genomics.deepvariant.LabelingMetrics.n_false_positive_sites', index=9,
      number=10, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='n_false_positive_alleles', full_name='learning.genomics.deepvariant.LabelingMetrics.n_false_positive_alleles', index=10,
      number=11, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='n_inexact_position_matches', full_name='learning.genomics.deepvariant.LabelingMetrics.n_inexact_position_matches', index=11,
      number=12, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='n_exact_position_matches', full_name='learning.genomics.deepvariant.LabelingMetrics.n_exact_position_matches', index=12,
      number=13, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='n_exact_position_and_allele_matches', full_name='learning.genomics.deepvariant.LabelingMetrics.n_exact_position_and_allele_matches', index=13,
      number=14, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='n_exact_position_and_allele_and_genotype_matches', full_name='learning.genomics.deepvariant.LabelingMetrics.n_exact_position_and_allele_and_genotype_matches', index=14,
      number=15, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='n_truth_multiallelics_sites_with_missed_alleles', full_name='learning.genomics.deepvariant.LabelingMetrics.n_truth_multiallelics_sites_with_missed_alleles', index=15,
      number=16, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=7991,
  serialized_end=8619,
)


_MAKEEXAMPLESSTATS = _descriptor.Descriptor(
  name='MakeExamplesStats',
  full_name='learning.genomics.deepvariant.MakeExamplesStats',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='num_examples', full_name='learning.genomics.deepvariant.MakeExamplesStats.num_examples', index=0,
      number=1, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='num_indels', full_name='learning.genomics.deepvariant.MakeExamplesStats.num_indels', index=1,
      number=2, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='num_snps', full_name='learning.genomics.deepvariant.MakeExamplesStats.num_snps', index=2,
      number=3, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='num_class_0', full_name='learning.genomics.deepvariant.MakeExamplesStats.num_class_0', index=3,
      number=4, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='num_class_1', full_name='learning.genomics.deepvariant.MakeExamplesStats.num_class_1', index=4,
      number=5, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='num_class_2', full_name='learning.genomics.deepvariant.MakeExamplesStats.num_class_2', index=5,
      number=6, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=8622,
  serialized_end=8764,
)


_MAKEEXAMPLESRUNINFO = _descriptor.Descriptor(
  name='MakeExamplesRunInfo',
  full_name='learning.genomics.deepvariant.MakeExamplesRunInfo',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='options', full_name='learning.genomics.deepvariant.MakeExamplesRunInfo.options', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='labeling_metrics', full_name='learning.genomics.deepvariant.MakeExamplesRunInfo.labeling_metrics', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='resource_metrics', full_name='learning.genomics.deepvariant.MakeExamplesRunInfo.resource_metrics', index=2,
      number=3, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='stats', full_name='learning.genomics.deepvariant.MakeExamplesRunInfo.stats', index=3,
      number=4, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=8767,
  serialized_end=9070,
)

_ALLELE.fields_by_name['type'].enum_type = _ALLELETYPE
_ALLELECOUNT_READALLELESENTRY.fields_by_name['value'].message_type = _ALLELE
_ALLELECOUNT_READALLELESENTRY.containing_type = _ALLELECOUNT
_ALLELECOUNT_ALLELES.fields_by_name['alleles'].message_type = _ALLELE
_ALLELECOUNT_ALLELES.containing_type = _ALLELECOUNT
_ALLELECOUNT_SAMPLEALLELESENTRY.fields_by_name['value'].message_type = _ALLELECOUNT_ALLELES
_ALLELECOUNT_SAMPLEALLELESENTRY.containing_type = _ALLELECOUNT
_ALLELECOUNT.fields_by_name['position'].message_type = third__party_dot_nucleus_dot_protos_dot_position__pb2._POSITION
_ALLELECOUNT.fields_by_name['read_alleles'].message_type = _ALLELECOUNT_READALLELESENTRY
_ALLELECOUNT.fields_by_name['sample_alleles'].message_type = _ALLELECOUNT_SAMPLEALLELESENTRY
_DEEPVARIANTCALL_SUPPORTINGREADS.containing_type = _DEEPVARIANTCALL
_DEEPVARIANTCALL_ALLELESUPPORTENTRY.fields_by_name['value'].message_type = _DEEPVARIANTCALL_SUPPORTINGREADS
_DEEPVARIANTCALL_ALLELESUPPORTENTRY.containing_type = _DEEPVARIANTCALL
_DEEPVARIANTCALL_ALLELEFREQUENCYENTRY.containing_type = _DEEPVARIANTCALL
_DEEPVARIANTCALL_READSUPPORT.containing_type = _DEEPVARIANTCALL
_DEEPVARIANTCALL_SUPPORTINGREADSEXT.fields_by_name['read_infos'].message_type = _DEEPVARIANTCALL_READSUPPORT
_DEEPVARIANTCALL_SUPPORTINGREADSEXT.containing_type = _DEEPVARIANTCALL
_DEEPVARIANTCALL_ALLELESUPPORTEXTENTRY.fields_by_name['value'].message_type = _DEEPVARIANTCALL_SUPPORTINGREADSEXT
_DEEPVARIANTCALL_ALLELESUPPORTEXTENTRY.containing_type = _DEEPVARIANTCALL
_DEEPVARIANTCALL.fields_by_name['variant'].message_type = third__party_dot_nucleus_dot_protos_dot_variants__pb2._VARIANT
_DEEPVARIANTCALL.fields_by_name['allele_support'].message_type = _DEEPVARIANTCALL_ALLELESUPPORTENTRY
_DEEPVARIANTCALL.fields_by_name['allele_frequency'].message_type = _DEEPVARIANTCALL_ALLELEFREQUENCYENTRY
_DEEPVARIANTCALL.fields_by_name['allele_support_ext'].message_type = _DEEPVARIANTCALL_ALLELESUPPORTEXTENTRY
_DEEPVARIANTCALL.fields_by_name['ref_support_ext'].message_type = _DEEPVARIANTCALL_SUPPORTINGREADSEXT
_ALLELECOUNTEROPTIONS.fields_by_name['read_requirements'].message_type = third__party_dot_nucleus_dot_protos_dot_reads__pb2._READREQUIREMENTS
_CALLVARIANTSOUTPUT_ALTALLELEINDICES.containing_type = _CALLVARIANTSOUTPUT
_CALLVARIANTSOUTPUT_DEBUGINFO.containing_type = _CALLVARIANTSOUTPUT
_CALLVARIANTSOUTPUT.fields_by_name['variant'].message_type = third__party_dot_nucleus_dot_protos_dot_variants__pb2._VARIANT
_CALLVARIANTSOUTPUT.fields_by_name['alt_allele_indices'].message_type = _CALLVARIANTSOUTPUT_ALTALLELEINDICES
_CALLVARIANTSOUTPUT.fields_by_name['debug_info'].message_type = _CALLVARIANTSOUTPUT_DEBUGINFO
_PILEUPIMAGEOPTIONS.fields_by_name['read_requirements'].message_type = third__party_dot_nucleus_dot_protos_dot_reads__pb2._READREQUIREMENTS
_PILEUPIMAGEOPTIONS.fields_by_name['multi_allelic_mode'].enum_type = _PILEUPIMAGEOPTIONS_MULTIALLELICMODE
_PILEUPIMAGEOPTIONS.fields_by_name['sequencing_type'].enum_type = _PILEUPIMAGEOPTIONS_SEQUENCINGTYPE
_PILEUPIMAGEOPTIONS_MULTIALLELICMODE.containing_type = _PILEUPIMAGEOPTIONS
_PILEUPIMAGEOPTIONS_SEQUENCINGTYPE.containing_type = _PILEUPIMAGEOPTIONS
_SAMPLEOPTIONS.fields_by_name['variant_caller_options'].message_type = _VARIANTCALLEROPTIONS
_MAKEEXAMPLESOPTIONS.fields_by_name['allele_counter_options'].message_type = _ALLELECOUNTEROPTIONS
_MAKEEXAMPLESOPTIONS.fields_by_name['deprecated_variant_caller_options'].message_type = _VARIANTCALLEROPTIONS
_MAKEEXAMPLESOPTIONS.fields_by_name['pic_options'].message_type = _PILEUPIMAGEOPTIONS
_MAKEEXAMPLESOPTIONS.fields_by_name['labeler_options'].message_type = _VARIANTLABELEROPTIONS
_MAKEEXAMPLESOPTIONS.fields_by_name['read_requirements'].message_type = third__party_dot_nucleus_dot_protos_dot_reads__pb2._READREQUIREMENTS
_MAKEEXAMPLESOPTIONS.fields_by_name['mode'].enum_type = _MAKEEXAMPLESOPTIONS_MODE
_MAKEEXAMPLESOPTIONS.fields_by_name['realigner_options'].message_type = deepvariant_dot_protos_dot_realigner__pb2._REALIGNEROPTIONS
_MAKEEXAMPLESOPTIONS.fields_by_name['labeler_algorithm'].enum_type = _MAKEEXAMPLESOPTIONS_LABELERALGORITHM
_MAKEEXAMPLESOPTIONS.fields_by_name['variant_caller'].enum_type = _MAKEEXAMPLESOPTIONS_VARIANTCALLER
_MAKEEXAMPLESOPTIONS.fields_by_name['sample_options'].message_type = _SAMPLEOPTIONS
_MAKEEXAMPLESOPTIONS_MODE.containing_type = _MAKEEXAMPLESOPTIONS
_MAKEEXAMPLESOPTIONS_LABELERALGORITHM.containing_type = _MAKEEXAMPLESOPTIONS
_MAKEEXAMPLESOPTIONS_VARIANTCALLER.containing_type = _MAKEEXAMPLESOPTIONS
_DEEPVARIANTINPUTDATASETS_DATASET.containing_type = _DEEPVARIANTINPUTDATASETS
_DEEPVARIANTINPUTDATASETS.fields_by_name['datasets'].message_type = _DEEPVARIANTINPUTDATASETS_DATASET
_DEEPVARIANTINPUTDATASETSGROUP.fields_by_name['groups'].message_type = _DEEPVARIANTINPUTDATASETS
_MAKEEXAMPLESRUNINFO.fields_by_name['options'].message_type = _MAKEEXAMPLESOPTIONS
_MAKEEXAMPLESRUNINFO.fields_by_name['labeling_metrics'].message_type = _LABELINGMETRICS
_MAKEEXAMPLESRUNINFO.fields_by_name['resource_metrics'].message_type = deepvariant_dot_protos_dot_resources__pb2._RESOURCEMETRICS
_MAKEEXAMPLESRUNINFO.fields_by_name['stats'].message_type = _MAKEEXAMPLESSTATS
DESCRIPTOR.message_types_by_name['Allele'] = _ALLELE
DESCRIPTOR.message_types_by_name['AlleleCount'] = _ALLELECOUNT
DESCRIPTOR.message_types_by_name['AlleleCountSummary'] = _ALLELECOUNTSUMMARY
DESCRIPTOR.message_types_by_name['DeepVariantCall'] = _DEEPVARIANTCALL
DESCRIPTOR.message_types_by_name['AlleleCounterOptions'] = _ALLELECOUNTEROPTIONS
DESCRIPTOR.message_types_by_name['CallVariantsOutput'] = _CALLVARIANTSOUTPUT
DESCRIPTOR.message_types_by_name['VariantCallerOptions'] = _VARIANTCALLEROPTIONS
DESCRIPTOR.message_types_by_name['VariantLabelerOptions'] = _VARIANTLABELEROPTIONS
DESCRIPTOR.message_types_by_name['PileupImageOptions'] = _PILEUPIMAGEOPTIONS
DESCRIPTOR.message_types_by_name['SampleOptions'] = _SAMPLEOPTIONS
DESCRIPTOR.message_types_by_name['MakeExamplesOptions'] = _MAKEEXAMPLESOPTIONS
DESCRIPTOR.message_types_by_name['DeepVariantDatasetConfig'] = _DEEPVARIANTDATASETCONFIG
DESCRIPTOR.message_types_by_name['DeepVariantInputDatasets'] = _DEEPVARIANTINPUTDATASETS
DESCRIPTOR.message_types_by_name['DeepVariantInputDatasetsGroup'] = _DEEPVARIANTINPUTDATASETSGROUP
DESCRIPTOR.message_types_by_name['LabelingMetrics'] = _LABELINGMETRICS
DESCRIPTOR.message_types_by_name['MakeExamplesStats'] = _MAKEEXAMPLESSTATS
DESCRIPTOR.message_types_by_name['MakeExamplesRunInfo'] = _MAKEEXAMPLESRUNINFO
DESCRIPTOR.enum_types_by_name['AlleleType'] = _ALLELETYPE
_sym_db.RegisterFileDescriptor(DESCRIPTOR)

Allele = _reflection.GeneratedProtocolMessageType('Allele', (_message.Message,), {
  'DESCRIPTOR' : _ALLELE,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.Allele)
  })
_sym_db.RegisterMessage(Allele)

AlleleCount = _reflection.GeneratedProtocolMessageType('AlleleCount', (_message.Message,), {

  'ReadAllelesEntry' : _reflection.GeneratedProtocolMessageType('ReadAllelesEntry', (_message.Message,), {
    'DESCRIPTOR' : _ALLELECOUNT_READALLELESENTRY,
    '__module__' : 'deepvariant.protos.deepvariant_pb2'
    # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.AlleleCount.ReadAllelesEntry)
    })
  ,

  'Alleles' : _reflection.GeneratedProtocolMessageType('Alleles', (_message.Message,), {
    'DESCRIPTOR' : _ALLELECOUNT_ALLELES,
    '__module__' : 'deepvariant.protos.deepvariant_pb2'
    # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.AlleleCount.Alleles)
    })
  ,

  'SampleAllelesEntry' : _reflection.GeneratedProtocolMessageType('SampleAllelesEntry', (_message.Message,), {
    'DESCRIPTOR' : _ALLELECOUNT_SAMPLEALLELESENTRY,
    '__module__' : 'deepvariant.protos.deepvariant_pb2'
    # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.AlleleCount.SampleAllelesEntry)
    })
  ,
  'DESCRIPTOR' : _ALLELECOUNT,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.AlleleCount)
  })
_sym_db.RegisterMessage(AlleleCount)
_sym_db.RegisterMessage(AlleleCount.ReadAllelesEntry)
_sym_db.RegisterMessage(AlleleCount.Alleles)
_sym_db.RegisterMessage(AlleleCount.SampleAllelesEntry)

AlleleCountSummary = _reflection.GeneratedProtocolMessageType('AlleleCountSummary', (_message.Message,), {
  'DESCRIPTOR' : _ALLELECOUNTSUMMARY,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.AlleleCountSummary)
  })
_sym_db.RegisterMessage(AlleleCountSummary)

DeepVariantCall = _reflection.GeneratedProtocolMessageType('DeepVariantCall', (_message.Message,), {

  'SupportingReads' : _reflection.GeneratedProtocolMessageType('SupportingReads', (_message.Message,), {
    'DESCRIPTOR' : _DEEPVARIANTCALL_SUPPORTINGREADS,
    '__module__' : 'deepvariant.protos.deepvariant_pb2'
    # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.DeepVariantCall.SupportingReads)
    })
  ,

  'AlleleSupportEntry' : _reflection.GeneratedProtocolMessageType('AlleleSupportEntry', (_message.Message,), {
    'DESCRIPTOR' : _DEEPVARIANTCALL_ALLELESUPPORTENTRY,
    '__module__' : 'deepvariant.protos.deepvariant_pb2'
    # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.DeepVariantCall.AlleleSupportEntry)
    })
  ,

  'AlleleFrequencyEntry' : _reflection.GeneratedProtocolMessageType('AlleleFrequencyEntry', (_message.Message,), {
    'DESCRIPTOR' : _DEEPVARIANTCALL_ALLELEFREQUENCYENTRY,
    '__module__' : 'deepvariant.protos.deepvariant_pb2'
    # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.DeepVariantCall.AlleleFrequencyEntry)
    })
  ,

  'ReadSupport' : _reflection.GeneratedProtocolMessageType('ReadSupport', (_message.Message,), {
    'DESCRIPTOR' : _DEEPVARIANTCALL_READSUPPORT,
    '__module__' : 'deepvariant.protos.deepvariant_pb2'
    # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.DeepVariantCall.ReadSupport)
    })
  ,

  'SupportingReadsExt' : _reflection.GeneratedProtocolMessageType('SupportingReadsExt', (_message.Message,), {
    'DESCRIPTOR' : _DEEPVARIANTCALL_SUPPORTINGREADSEXT,
    '__module__' : 'deepvariant.protos.deepvariant_pb2'
    # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.DeepVariantCall.SupportingReadsExt)
    })
  ,

  'AlleleSupportExtEntry' : _reflection.GeneratedProtocolMessageType('AlleleSupportExtEntry', (_message.Message,), {
    'DESCRIPTOR' : _DEEPVARIANTCALL_ALLELESUPPORTEXTENTRY,
    '__module__' : 'deepvariant.protos.deepvariant_pb2'
    # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.DeepVariantCall.AlleleSupportExtEntry)
    })
  ,
  'DESCRIPTOR' : _DEEPVARIANTCALL,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.DeepVariantCall)
  })
_sym_db.RegisterMessage(DeepVariantCall)
_sym_db.RegisterMessage(DeepVariantCall.SupportingReads)
_sym_db.RegisterMessage(DeepVariantCall.AlleleSupportEntry)
_sym_db.RegisterMessage(DeepVariantCall.AlleleFrequencyEntry)
_sym_db.RegisterMessage(DeepVariantCall.ReadSupport)
_sym_db.RegisterMessage(DeepVariantCall.SupportingReadsExt)
_sym_db.RegisterMessage(DeepVariantCall.AlleleSupportExtEntry)

AlleleCounterOptions = _reflection.GeneratedProtocolMessageType('AlleleCounterOptions', (_message.Message,), {
  'DESCRIPTOR' : _ALLELECOUNTEROPTIONS,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.AlleleCounterOptions)
  })
_sym_db.RegisterMessage(AlleleCounterOptions)

CallVariantsOutput = _reflection.GeneratedProtocolMessageType('CallVariantsOutput', (_message.Message,), {

  'AltAlleleIndices' : _reflection.GeneratedProtocolMessageType('AltAlleleIndices', (_message.Message,), {
    'DESCRIPTOR' : _CALLVARIANTSOUTPUT_ALTALLELEINDICES,
    '__module__' : 'deepvariant.protos.deepvariant_pb2'
    # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.CallVariantsOutput.AltAlleleIndices)
    })
  ,

  'DebugInfo' : _reflection.GeneratedProtocolMessageType('DebugInfo', (_message.Message,), {
    'DESCRIPTOR' : _CALLVARIANTSOUTPUT_DEBUGINFO,
    '__module__' : 'deepvariant.protos.deepvariant_pb2'
    # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.CallVariantsOutput.DebugInfo)
    })
  ,
  'DESCRIPTOR' : _CALLVARIANTSOUTPUT,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.CallVariantsOutput)
  })
_sym_db.RegisterMessage(CallVariantsOutput)
_sym_db.RegisterMessage(CallVariantsOutput.AltAlleleIndices)
_sym_db.RegisterMessage(CallVariantsOutput.DebugInfo)

VariantCallerOptions = _reflection.GeneratedProtocolMessageType('VariantCallerOptions', (_message.Message,), {
  'DESCRIPTOR' : _VARIANTCALLEROPTIONS,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.VariantCallerOptions)
  })
_sym_db.RegisterMessage(VariantCallerOptions)

VariantLabelerOptions = _reflection.GeneratedProtocolMessageType('VariantLabelerOptions', (_message.Message,), {
  'DESCRIPTOR' : _VARIANTLABELEROPTIONS,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.VariantLabelerOptions)
  })
_sym_db.RegisterMessage(VariantLabelerOptions)

PileupImageOptions = _reflection.GeneratedProtocolMessageType('PileupImageOptions', (_message.Message,), {
  'DESCRIPTOR' : _PILEUPIMAGEOPTIONS,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.PileupImageOptions)
  })
_sym_db.RegisterMessage(PileupImageOptions)

SampleOptions = _reflection.GeneratedProtocolMessageType('SampleOptions', (_message.Message,), {
  'DESCRIPTOR' : _SAMPLEOPTIONS,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.SampleOptions)
  })
_sym_db.RegisterMessage(SampleOptions)

MakeExamplesOptions = _reflection.GeneratedProtocolMessageType('MakeExamplesOptions', (_message.Message,), {
  'DESCRIPTOR' : _MAKEEXAMPLESOPTIONS,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.MakeExamplesOptions)
  })
_sym_db.RegisterMessage(MakeExamplesOptions)

DeepVariantDatasetConfig = _reflection.GeneratedProtocolMessageType('DeepVariantDatasetConfig', (_message.Message,), {
  'DESCRIPTOR' : _DEEPVARIANTDATASETCONFIG,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.DeepVariantDatasetConfig)
  })
_sym_db.RegisterMessage(DeepVariantDatasetConfig)

DeepVariantInputDatasets = _reflection.GeneratedProtocolMessageType('DeepVariantInputDatasets', (_message.Message,), {

  'Dataset' : _reflection.GeneratedProtocolMessageType('Dataset', (_message.Message,), {
    'DESCRIPTOR' : _DEEPVARIANTINPUTDATASETS_DATASET,
    '__module__' : 'deepvariant.protos.deepvariant_pb2'
    # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.DeepVariantInputDatasets.Dataset)
    })
  ,
  'DESCRIPTOR' : _DEEPVARIANTINPUTDATASETS,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.DeepVariantInputDatasets)
  })
_sym_db.RegisterMessage(DeepVariantInputDatasets)
_sym_db.RegisterMessage(DeepVariantInputDatasets.Dataset)

DeepVariantInputDatasetsGroup = _reflection.GeneratedProtocolMessageType('DeepVariantInputDatasetsGroup', (_message.Message,), {
  'DESCRIPTOR' : _DEEPVARIANTINPUTDATASETSGROUP,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.DeepVariantInputDatasetsGroup)
  })
_sym_db.RegisterMessage(DeepVariantInputDatasetsGroup)

LabelingMetrics = _reflection.GeneratedProtocolMessageType('LabelingMetrics', (_message.Message,), {
  'DESCRIPTOR' : _LABELINGMETRICS,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.LabelingMetrics)
  })
_sym_db.RegisterMessage(LabelingMetrics)

MakeExamplesStats = _reflection.GeneratedProtocolMessageType('MakeExamplesStats', (_message.Message,), {
  'DESCRIPTOR' : _MAKEEXAMPLESSTATS,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.MakeExamplesStats)
  })
_sym_db.RegisterMessage(MakeExamplesStats)

MakeExamplesRunInfo = _reflection.GeneratedProtocolMessageType('MakeExamplesRunInfo', (_message.Message,), {
  'DESCRIPTOR' : _MAKEEXAMPLESRUNINFO,
  '__module__' : 'deepvariant.protos.deepvariant_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.MakeExamplesRunInfo)
  })
_sym_db.RegisterMessage(MakeExamplesRunInfo)


_ALLELECOUNT_READALLELESENTRY._options = None
_ALLELECOUNT_SAMPLEALLELESENTRY._options = None
_DEEPVARIANTCALL_ALLELESUPPORTENTRY._options = None
_DEEPVARIANTCALL_ALLELEFREQUENCYENTRY._options = None
_DEEPVARIANTCALL_ALLELESUPPORTEXTENTRY._options = None
_PILEUPIMAGEOPTIONS.fields_by_name['sort_by_haplotypes_sample_hp_tag']._options = None
_MAKEEXAMPLESOPTIONS.fields_by_name['deprecated_proposed_variants_filename']._options = None
_DEEPVARIANTINPUTDATASETS_DATASET.fields_by_name['sort_by_haplotypes_sample_hp_tag']._options = None
# @@protoc_insertion_point(module_scope)