# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: deepvariant/protos/realigner.proto

import sys
_b=sys.version_info[0]<3 and (lambda x:x) or (lambda x:x.encode('latin1'))
from google.protobuf import descriptor as _descriptor
from google.protobuf import message as _message
from google.protobuf import reflection as _reflection
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()


from .third_party.nucleus.protos import range_pb2 as third__party_dot_nucleus_dot_protos_dot_range__pb2


DESCRIPTOR = _descriptor.FileDescriptor(
  name='deepvariant/protos/realigner.proto',
  package='learning.genomics.deepvariant',
  syntax='proto3',
  serialized_options=None,
  serialized_pb=_b('\n\"deepvariant/protos/realigner.proto\x12\x1dlearning.genomics.deepvariant\x1a&third_party/nucleus/protos/range.proto\"S\n\x13\x43\x61ndidateHaplotypes\x12(\n\x04span\x18\x01 \x01(\x0b\x32\x1a.nucleus.genomics.v1.Range\x12\x12\n\nhaplotypes\x18\x02 \x03(\t\"\xbb\x05\n\x13WindowSelectorModel\x12P\n\nmodel_type\x18\x01 \x01(\x0e\x32<.learning.genomics.deepvariant.WindowSelectorModel.ModelType\x12l\n\x13variant_reads_model\x18\x02 \x01(\x0b\x32M.learning.genomics.deepvariant.WindowSelectorModel.VariantReadsThresholdModelH\x00\x12n\n\x19\x61llele_count_linear_model\x18\x03 \x01(\x0b\x32I.learning.genomics.deepvariant.WindowSelectorModel.AlleleCountLinearModelH\x00\x1a`\n\x1aVariantReadsThresholdModel\x12 \n\x18min_num_supporting_reads\x18\x01 \x01(\x05\x12 \n\x18max_num_supporting_reads\x18\x02 \x01(\x05\x1a\xc0\x01\n\x16\x41lleleCountLinearModel\x12\x0c\n\x04\x62ias\x18\x01 \x01(\x02\x12\x17\n\x0f\x63oeff_soft_clip\x18\x02 \x01(\x02\x12\x1a\n\x12\x63oeff_substitution\x18\x03 \x01(\x02\x12\x17\n\x0f\x63oeff_insertion\x18\x04 \x01(\x02\x12\x16\n\x0e\x63oeff_deletion\x18\x05 \x01(\x02\x12\x17\n\x0f\x63oeff_reference\x18\x06 \x01(\x02\x12\x19\n\x11\x64\x65\x63ision_boundary\x18\x07 \x01(\x02\"F\n\tModelType\x12\r\n\tUNDEFINED\x10\x00\x12\x11\n\rVARIANT_READS\x10\x01\x12\x17\n\x13\x41LLELE_COUNT_LINEAR\x10\x02\x42\x07\n\x05model\"\xcf\x02\n\x15WindowSelectorOptions\x12 \n\x18min_num_supporting_reads\x18\x01 \x01(\x05\x12 \n\x18max_num_supporting_reads\x18\x02 \x01(\x05\x12\x10\n\x08min_mapq\x18\x03 \x01(\x05\x12\x18\n\x10min_base_quality\x18\x04 \x01(\x05\x12\x1c\n\x14min_windows_distance\x18\x05 \x01(\x05\x12\x17\n\x0fmax_window_size\x18\x06 \x01(\x05\x12\x1e\n\x16region_expansion_in_bp\x18\x07 \x01(\x05\x12Q\n\x15window_selector_model\x18\x08 \x01(\x0b\x32\x32.learning.genomics.deepvariant.WindowSelectorModel\x12\x1c\n\x14keep_legacy_behavior\x18\t \x01(\x08\"\xa0\x01\n\x14\x44\x65\x42ruijnGraphOptions\x12\r\n\x05min_k\x18\x01 \x01(\x05\x12\r\n\x05max_k\x18\x02 \x01(\x05\x12\x0e\n\x06step_k\x18\x03 \x01(\x05\x12\x10\n\x08min_mapq\x18\x04 \x01(\x05\x12\x18\n\x10min_base_quality\x18\x05 \x01(\x05\x12\x17\n\x0fmin_edge_weight\x18\x06 \x01(\x05\x12\x15\n\rmax_num_paths\x18\x07 \x01(\x05\"\xfe\x01\n\x0e\x41lignerOptions\x12\r\n\x05match\x18\x01 \x01(\x05\x12\x10\n\x08mismatch\x18\x02 \x01(\x05\x12\x10\n\x08gap_open\x18\x03 \x01(\x05\x12\x12\n\ngap_extend\x18\x04 \x01(\x05\x12\t\n\x01k\x18\x05 \x01(\x05\x12\x12\n\nerror_rate\x18\x06 \x01(\x02\x12\x11\n\tread_size\x18\x08 \x01(\x05\x12\x11\n\tkmer_size\x18\t \x01(\x05\x12\x1d\n\x15max_num_of_mismatches\x18\n \x01(\x05\x12(\n realignment_similarity_threshold\x18\x0b \x01(\x01\x12\x17\n\x0f\x66orce_alignment\x18\x0c \x01(\x08\"Q\n\x0b\x44iagnostics\x12\x0f\n\x07\x65nabled\x18\x01 \x01(\x08\x12\x13\n\x0boutput_root\x18\x02 \x01(\t\x12\x1c\n\x14\x65mit_realigned_reads\x18\x03 \x01(\x08\"\xdb\x02\n\x10RealignerOptions\x12G\n\tws_config\x18\x01 \x01(\x0b\x32\x34.learning.genomics.deepvariant.WindowSelectorOptions\x12G\n\ndbg_config\x18\x02 \x01(\x0b\x32\x33.learning.genomics.deepvariant.DeBruijnGraphOptions\x12\x41\n\naln_config\x18\x03 \x01(\x0b\x32-.learning.genomics.deepvariant.AlignerOptions\x12?\n\x0b\x64iagnostics\x18\x04 \x01(\x0b\x32*.learning.genomics.deepvariant.Diagnostics\x12\x18\n\x10split_skip_reads\x18\x05 \x01(\x08\x12\x17\n\x0fnormalize_reads\x18\x06 \x01(\x08\x62\x06proto3')
  ,
  dependencies=[third__party_dot_nucleus_dot_protos_dot_range__pb2.DESCRIPTOR,])



_WINDOWSELECTORMODEL_MODELTYPE = _descriptor.EnumDescriptor(
  name='ModelType',
  full_name='learning.genomics.deepvariant.WindowSelectorModel.ModelType',
  filename=None,
  file=DESCRIPTOR,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNDEFINED', index=0, number=0,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='VARIANT_READS', index=1, number=1,
      serialized_options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='ALLELE_COUNT_LINEAR', index=2, number=2,
      serialized_options=None,
      type=None),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=815,
  serialized_end=885,
)
_sym_db.RegisterEnumDescriptor(_WINDOWSELECTORMODEL_MODELTYPE)


_CANDIDATEHAPLOTYPES = _descriptor.Descriptor(
  name='CandidateHaplotypes',
  full_name='learning.genomics.deepvariant.CandidateHaplotypes',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='span', full_name='learning.genomics.deepvariant.CandidateHaplotypes.span', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='haplotypes', full_name='learning.genomics.deepvariant.CandidateHaplotypes.haplotypes', index=1,
      number=2, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=109,
  serialized_end=192,
)


_WINDOWSELECTORMODEL_VARIANTREADSTHRESHOLDMODEL = _descriptor.Descriptor(
  name='VariantReadsThresholdModel',
  full_name='learning.genomics.deepvariant.WindowSelectorModel.VariantReadsThresholdModel',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='min_num_supporting_reads', full_name='learning.genomics.deepvariant.WindowSelectorModel.VariantReadsThresholdModel.min_num_supporting_reads', index=0,
      number=1, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='max_num_supporting_reads', full_name='learning.genomics.deepvariant.WindowSelectorModel.VariantReadsThresholdModel.max_num_supporting_reads', index=1,
      number=2, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=522,
  serialized_end=618,
)

_WINDOWSELECTORMODEL_ALLELECOUNTLINEARMODEL = _descriptor.Descriptor(
  name='AlleleCountLinearModel',
  full_name='learning.genomics.deepvariant.WindowSelectorModel.AlleleCountLinearModel',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='bias', full_name='learning.genomics.deepvariant.WindowSelectorModel.AlleleCountLinearModel.bias', index=0,
      number=1, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='coeff_soft_clip', full_name='learning.genomics.deepvariant.WindowSelectorModel.AlleleCountLinearModel.coeff_soft_clip', index=1,
      number=2, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='coeff_substitution', full_name='learning.genomics.deepvariant.WindowSelectorModel.AlleleCountLinearModel.coeff_substitution', index=2,
      number=3, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='coeff_insertion', full_name='learning.genomics.deepvariant.WindowSelectorModel.AlleleCountLinearModel.coeff_insertion', index=3,
      number=4, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='coeff_deletion', full_name='learning.genomics.deepvariant.WindowSelectorModel.AlleleCountLinearModel.coeff_deletion', index=4,
      number=5, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='coeff_reference', full_name='learning.genomics.deepvariant.WindowSelectorModel.AlleleCountLinearModel.coeff_reference', index=5,
      number=6, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='decision_boundary', full_name='learning.genomics.deepvariant.WindowSelectorModel.AlleleCountLinearModel.decision_boundary', index=6,
      number=7, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=621,
  serialized_end=813,
)

_WINDOWSELECTORMODEL = _descriptor.Descriptor(
  name='WindowSelectorModel',
  full_name='learning.genomics.deepvariant.WindowSelectorModel',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='model_type', full_name='learning.genomics.deepvariant.WindowSelectorModel.model_type', index=0,
      number=1, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='variant_reads_model', full_name='learning.genomics.deepvariant.WindowSelectorModel.variant_reads_model', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='allele_count_linear_model', full_name='learning.genomics.deepvariant.WindowSelectorModel.allele_count_linear_model', index=2,
      number=3, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[_WINDOWSELECTORMODEL_VARIANTREADSTHRESHOLDMODEL, _WINDOWSELECTORMODEL_ALLELECOUNTLINEARMODEL, ],
  enum_types=[
    _WINDOWSELECTORMODEL_MODELTYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
    _descriptor.OneofDescriptor(
      name='model', full_name='learning.genomics.deepvariant.WindowSelectorModel.model',
      index=0, containing_type=None, fields=[]),
  ],
  serialized_start=195,
  serialized_end=894,
)


_WINDOWSELECTOROPTIONS = _descriptor.Descriptor(
  name='WindowSelectorOptions',
  full_name='learning.genomics.deepvariant.WindowSelectorOptions',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='min_num_supporting_reads', full_name='learning.genomics.deepvariant.WindowSelectorOptions.min_num_supporting_reads', index=0,
      number=1, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='max_num_supporting_reads', full_name='learning.genomics.deepvariant.WindowSelectorOptions.max_num_supporting_reads', index=1,
      number=2, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='min_mapq', full_name='learning.genomics.deepvariant.WindowSelectorOptions.min_mapq', index=2,
      number=3, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='min_base_quality', full_name='learning.genomics.deepvariant.WindowSelectorOptions.min_base_quality', index=3,
      number=4, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='min_windows_distance', full_name='learning.genomics.deepvariant.WindowSelectorOptions.min_windows_distance', index=4,
      number=5, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='max_window_size', full_name='learning.genomics.deepvariant.WindowSelectorOptions.max_window_size', index=5,
      number=6, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='region_expansion_in_bp', full_name='learning.genomics.deepvariant.WindowSelectorOptions.region_expansion_in_bp', index=6,
      number=7, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='window_selector_model', full_name='learning.genomics.deepvariant.WindowSelectorOptions.window_selector_model', index=7,
      number=8, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='keep_legacy_behavior', full_name='learning.genomics.deepvariant.WindowSelectorOptions.keep_legacy_behavior', index=8,
      number=9, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=897,
  serialized_end=1232,
)


_DEBRUIJNGRAPHOPTIONS = _descriptor.Descriptor(
  name='DeBruijnGraphOptions',
  full_name='learning.genomics.deepvariant.DeBruijnGraphOptions',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='min_k', full_name='learning.genomics.deepvariant.DeBruijnGraphOptions.min_k', index=0,
      number=1, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='max_k', full_name='learning.genomics.deepvariant.DeBruijnGraphOptions.max_k', index=1,
      number=2, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='step_k', full_name='learning.genomics.deepvariant.DeBruijnGraphOptions.step_k', index=2,
      number=3, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='min_mapq', full_name='learning.genomics.deepvariant.DeBruijnGraphOptions.min_mapq', index=3,
      number=4, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='min_base_quality', full_name='learning.genomics.deepvariant.DeBruijnGraphOptions.min_base_quality', index=4,
      number=5, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='min_edge_weight', full_name='learning.genomics.deepvariant.DeBruijnGraphOptions.min_edge_weight', index=5,
      number=6, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='max_num_paths', full_name='learning.genomics.deepvariant.DeBruijnGraphOptions.max_num_paths', index=6,
      number=7, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=1235,
  serialized_end=1395,
)


_ALIGNEROPTIONS = _descriptor.Descriptor(
  name='AlignerOptions',
  full_name='learning.genomics.deepvariant.AlignerOptions',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='match', full_name='learning.genomics.deepvariant.AlignerOptions.match', index=0,
      number=1, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='mismatch', full_name='learning.genomics.deepvariant.AlignerOptions.mismatch', index=1,
      number=2, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='gap_open', full_name='learning.genomics.deepvariant.AlignerOptions.gap_open', index=2,
      number=3, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='gap_extend', full_name='learning.genomics.deepvariant.AlignerOptions.gap_extend', index=3,
      number=4, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='k', full_name='learning.genomics.deepvariant.AlignerOptions.k', index=4,
      number=5, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='error_rate', full_name='learning.genomics.deepvariant.AlignerOptions.error_rate', index=5,
      number=6, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='read_size', full_name='learning.genomics.deepvariant.AlignerOptions.read_size', index=6,
      number=8, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='kmer_size', full_name='learning.genomics.deepvariant.AlignerOptions.kmer_size', index=7,
      number=9, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='max_num_of_mismatches', full_name='learning.genomics.deepvariant.AlignerOptions.max_num_of_mismatches', index=8,
      number=10, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='realignment_similarity_threshold', full_name='learning.genomics.deepvariant.AlignerOptions.realignment_similarity_threshold', index=9,
      number=11, type=1, cpp_type=5, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='force_alignment', full_name='learning.genomics.deepvariant.AlignerOptions.force_alignment', index=10,
      number=12, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=1398,
  serialized_end=1652,
)


_DIAGNOSTICS = _descriptor.Descriptor(
  name='Diagnostics',
  full_name='learning.genomics.deepvariant.Diagnostics',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='enabled', full_name='learning.genomics.deepvariant.Diagnostics.enabled', index=0,
      number=1, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='output_root', full_name='learning.genomics.deepvariant.Diagnostics.output_root', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='emit_realigned_reads', full_name='learning.genomics.deepvariant.Diagnostics.emit_realigned_reads', index=2,
      number=3, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=1654,
  serialized_end=1735,
)


_REALIGNEROPTIONS = _descriptor.Descriptor(
  name='RealignerOptions',
  full_name='learning.genomics.deepvariant.RealignerOptions',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='ws_config', full_name='learning.genomics.deepvariant.RealignerOptions.ws_config', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='dbg_config', full_name='learning.genomics.deepvariant.RealignerOptions.dbg_config', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='aln_config', full_name='learning.genomics.deepvariant.RealignerOptions.aln_config', index=2,
      number=3, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='diagnostics', full_name='learning.genomics.deepvariant.RealignerOptions.diagnostics', index=3,
      number=4, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='split_skip_reads', full_name='learning.genomics.deepvariant.RealignerOptions.split_skip_reads', index=4,
      number=5, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
    _descriptor.FieldDescriptor(
      name='normalize_reads', full_name='learning.genomics.deepvariant.RealignerOptions.normalize_reads', index=5,
      number=6, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=1738,
  serialized_end=2085,
)

_CANDIDATEHAPLOTYPES.fields_by_name['span'].message_type = third__party_dot_nucleus_dot_protos_dot_range__pb2._RANGE
_WINDOWSELECTORMODEL_VARIANTREADSTHRESHOLDMODEL.containing_type = _WINDOWSELECTORMODEL
_WINDOWSELECTORMODEL_ALLELECOUNTLINEARMODEL.containing_type = _WINDOWSELECTORMODEL
_WINDOWSELECTORMODEL.fields_by_name['model_type'].enum_type = _WINDOWSELECTORMODEL_MODELTYPE
_WINDOWSELECTORMODEL.fields_by_name['variant_reads_model'].message_type = _WINDOWSELECTORMODEL_VARIANTREADSTHRESHOLDMODEL
_WINDOWSELECTORMODEL.fields_by_name['allele_count_linear_model'].message_type = _WINDOWSELECTORMODEL_ALLELECOUNTLINEARMODEL
_WINDOWSELECTORMODEL_MODELTYPE.containing_type = _WINDOWSELECTORMODEL
_WINDOWSELECTORMODEL.oneofs_by_name['model'].fields.append(
  _WINDOWSELECTORMODEL.fields_by_name['variant_reads_model'])
_WINDOWSELECTORMODEL.fields_by_name['variant_reads_model'].containing_oneof = _WINDOWSELECTORMODEL.oneofs_by_name['model']
_WINDOWSELECTORMODEL.oneofs_by_name['model'].fields.append(
  _WINDOWSELECTORMODEL.fields_by_name['allele_count_linear_model'])
_WINDOWSELECTORMODEL.fields_by_name['allele_count_linear_model'].containing_oneof = _WINDOWSELECTORMODEL.oneofs_by_name['model']
_WINDOWSELECTOROPTIONS.fields_by_name['window_selector_model'].message_type = _WINDOWSELECTORMODEL
_REALIGNEROPTIONS.fields_by_name['ws_config'].message_type = _WINDOWSELECTOROPTIONS
_REALIGNEROPTIONS.fields_by_name['dbg_config'].message_type = _DEBRUIJNGRAPHOPTIONS
_REALIGNEROPTIONS.fields_by_name['aln_config'].message_type = _ALIGNEROPTIONS
_REALIGNEROPTIONS.fields_by_name['diagnostics'].message_type = _DIAGNOSTICS
DESCRIPTOR.message_types_by_name['CandidateHaplotypes'] = _CANDIDATEHAPLOTYPES
DESCRIPTOR.message_types_by_name['WindowSelectorModel'] = _WINDOWSELECTORMODEL
DESCRIPTOR.message_types_by_name['WindowSelectorOptions'] = _WINDOWSELECTOROPTIONS
DESCRIPTOR.message_types_by_name['DeBruijnGraphOptions'] = _DEBRUIJNGRAPHOPTIONS
DESCRIPTOR.message_types_by_name['AlignerOptions'] = _ALIGNEROPTIONS
DESCRIPTOR.message_types_by_name['Diagnostics'] = _DIAGNOSTICS
DESCRIPTOR.message_types_by_name['RealignerOptions'] = _REALIGNEROPTIONS
_sym_db.RegisterFileDescriptor(DESCRIPTOR)

CandidateHaplotypes = _reflection.GeneratedProtocolMessageType('CandidateHaplotypes', (_message.Message,), {
  'DESCRIPTOR' : _CANDIDATEHAPLOTYPES,
  '__module__' : 'deepvariant.protos.realigner_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.CandidateHaplotypes)
  })
_sym_db.RegisterMessage(CandidateHaplotypes)

WindowSelectorModel = _reflection.GeneratedProtocolMessageType('WindowSelectorModel', (_message.Message,), {

  'VariantReadsThresholdModel' : _reflection.GeneratedProtocolMessageType('VariantReadsThresholdModel', (_message.Message,), {
    'DESCRIPTOR' : _WINDOWSELECTORMODEL_VARIANTREADSTHRESHOLDMODEL,
    '__module__' : 'deepvariant.protos.realigner_pb2'
    # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.WindowSelectorModel.VariantReadsThresholdModel)
    })
  ,

  'AlleleCountLinearModel' : _reflection.GeneratedProtocolMessageType('AlleleCountLinearModel', (_message.Message,), {
    'DESCRIPTOR' : _WINDOWSELECTORMODEL_ALLELECOUNTLINEARMODEL,
    '__module__' : 'deepvariant.protos.realigner_pb2'
    # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.WindowSelectorModel.AlleleCountLinearModel)
    })
  ,
  'DESCRIPTOR' : _WINDOWSELECTORMODEL,
  '__module__' : 'deepvariant.protos.realigner_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.WindowSelectorModel)
  })
_sym_db.RegisterMessage(WindowSelectorModel)
_sym_db.RegisterMessage(WindowSelectorModel.VariantReadsThresholdModel)
_sym_db.RegisterMessage(WindowSelectorModel.AlleleCountLinearModel)

WindowSelectorOptions = _reflection.GeneratedProtocolMessageType('WindowSelectorOptions', (_message.Message,), {
  'DESCRIPTOR' : _WINDOWSELECTOROPTIONS,
  '__module__' : 'deepvariant.protos.realigner_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.WindowSelectorOptions)
  })
_sym_db.RegisterMessage(WindowSelectorOptions)

DeBruijnGraphOptions = _reflection.GeneratedProtocolMessageType('DeBruijnGraphOptions', (_message.Message,), {
  'DESCRIPTOR' : _DEBRUIJNGRAPHOPTIONS,
  '__module__' : 'deepvariant.protos.realigner_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.DeBruijnGraphOptions)
  })
_sym_db.RegisterMessage(DeBruijnGraphOptions)

AlignerOptions = _reflection.GeneratedProtocolMessageType('AlignerOptions', (_message.Message,), {
  'DESCRIPTOR' : _ALIGNEROPTIONS,
  '__module__' : 'deepvariant.protos.realigner_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.AlignerOptions)
  })
_sym_db.RegisterMessage(AlignerOptions)

Diagnostics = _reflection.GeneratedProtocolMessageType('Diagnostics', (_message.Message,), {
  'DESCRIPTOR' : _DIAGNOSTICS,
  '__module__' : 'deepvariant.protos.realigner_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.Diagnostics)
  })
_sym_db.RegisterMessage(Diagnostics)

RealignerOptions = _reflection.GeneratedProtocolMessageType('RealignerOptions', (_message.Message,), {
  'DESCRIPTOR' : _REALIGNEROPTIONS,
  '__module__' : 'deepvariant.protos.realigner_pb2'
  # @@protoc_insertion_point(class_scope:learning.genomics.deepvariant.RealignerOptions)
  })
_sym_db.RegisterMessage(RealignerOptions)


# @@protoc_insertion_point(module_scope)
